/* CVS $Id: $ */
package com.genohm.boinq.generated.vocabularies; 
import org.apache.jena.ontology.AnnotationProperty;
import org.apache.jena.ontology.ObjectProperty;
import org.apache.jena.ontology.OntClass;
import org.apache.jena.ontology.OntModel;
import org.apache.jena.ontology.OntModelSpec;
import org.apache.jena.rdf.model.ModelFactory;
import org.apache.jena.rdf.model.Resource;
 
/**
 * Vocabulary definitions from ontologies/so.owl 
 * @author Auto-generated by schemagen on 09 Sep 2015 10:52 
 */
public class SoVocab {
    /** <p>The ontology model that holds the vocabulary terms</p> */
    private static OntModel m_model = ModelFactory.createOntologyModel( OntModelSpec.OWL_MEM, null );
    
    /** <p>The namespace of the vocabulary as a string</p> */
    public static final String NS = "http://www.w3.org/2002/07/owl#";
    
    /** <p>The namespace of the vocabulary as a string</p>
     *  @see #NS */
    public static String getURI() {return NS;}
    
    /** <p>The namespace of the vocabulary as a resource</p> */
    public static final Resource NAMESPACE = m_model.createResource( NS );
    
    public static final ObjectProperty adjacent_to = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#adjacent_to" );
    
    /** <p>This relationship is vague and up for discussion.</p> */
    public static final ObjectProperty associated_with = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#associated_with" );
    
    /** <p>If A is a feature with multiple regions such as a multi exon transcript, the 
     *  supporting EST evidence is complete if each of the regions is supported by 
     *  an equivalent region in B. Also there must be no extra regions in B that are 
     *  not represented in A. This relationship was requested by jeltje on the SO 
     *  term tracker. The thread for the discussion is available can be accessed via 
     *  tracker ID:1917222.</p>
     */
    public static final ObjectProperty complete_evidence_for_feature = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#complete_evidence_for_feature" );
    
    /** <p>Example: A splice_junction connects_on exon, exon, mature_transcript.</p> */
    public static final ObjectProperty connects_on = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#connects_on" );
    
    /** <p>The inverse is contains. Example: intein contained_by immature_peptide_region.</p> */
    public static final ObjectProperty contained_by = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#contained_by" );
    
    /** <p>Example: pre_miRNA contains miRNA_loop.</p> */
    public static final ObjectProperty contains = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#contains" );
    
    public static final ObjectProperty derives_from = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#derives_from" );
    
    public static final ObjectProperty disconnected_from = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#disconnected_from" );
    
    public static final ObjectProperty edited_from = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#edited_from" );
    
    public static final ObjectProperty edited_to = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#edited_to" );
    
    /** <p>This relationship was requested by nlw on the SO term tracker. The thread 
     *  for the discussion is available can be accessed via tracker ID:1917222.</p>
     */
    public static final ObjectProperty evidence_for_feature = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#evidence_for_feature" );
    
    /** <p>Tracker id: 2594157.</p> */
    public static final ObjectProperty exemplar_of = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#exemplar_of" );
    
    /** <p>Example CDS finished_by stop_codon.</p> */
    public static final ObjectProperty finished_by = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#finished_by" );
    
    /** <p>Example: stop_codon finishes CDS.</p> */
    public static final ObjectProperty finishes = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#finishes" );
    
    /** <p>A relation with which to annotate the changes in a variant sequence with respect 
     *  to a reference. For example a variant transcript may gain a stop codon not 
     *  present in the reference sequence.</p>
     */
    public static final ObjectProperty gained = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#gained" );
    
    public static final ObjectProperty genome_of = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#genome_of" );
    
    public static final ObjectProperty guided_by = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#guided_by" );
    
    public static final ObjectProperty guides = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#guides" );
    
    /** <p>Example: mRNA has_integral_part CDS.</p> */
    public static final ObjectProperty has_integral_part = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#has_integral_part" );
    
    public static final ObjectProperty has_origin = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#has_origin" );
    
    /** <p>Example: operon has_part gene.</p> */
    public static final ObjectProperty has_part = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#has_part" );
    
    /** <p>The relationship between a feature and an attribute.</p> */
    public static final ObjectProperty has_quality = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#has_quality" );
    
    public static final ObjectProperty homologous_to = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#homologous_to" );
    
    /** <p>Example: exon integral_part_of transcript.</p> */
    public static final ObjectProperty integral_part_of = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#integral_part_of" );
    
    /** <p>Example: region is consecutive_sequence of base.</p> */
    public static final ObjectProperty is_consecutive_sequence_of = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#is_consecutive_sequence_of" );
    
    /** <p>A relation with which to annotate the changes in a variant sequence with respect 
     *  to a reference. For example a variant transcript may have lost a stop codon 
     *  present in the reference sequence.</p>
     */
    public static final ObjectProperty lost = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#lost" );
    
    /** <p>Example: non_coding_region_of_exon maximally_overlaps the intersections of 
     *  exon and UTR.</p>
     */
    public static final ObjectProperty maximally_overlaps = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#maximally_overlaps" );
    
    /** <p>A subtype of part_of. Inverse is collection_of. Winston, M, Chaffin, R, Herrmann: 
     *  A taxonomy of part-whole relations. Cognitive Science 1987, 11:417-444.</p>
     */
    public static final ObjectProperty member_of = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#member_of" );
    
    public static final ObjectProperty non_functional_homolog_of = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#non_functional_homolog_of" );
    
    public static final ObjectProperty orthologous_to = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#orthologous_to" );
    
    /** <p>Example: coding_exon overlaps CDS.</p> */
    public static final ObjectProperty overlaps = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#overlaps" );
    
    public static final ObjectProperty paralogous_to = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#paralogous_to" );
    
    /** <p>Example: amino_acid part_of polypeptide.</p> */
    public static final ObjectProperty part_of = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#part_of" );
    
    public static final ObjectProperty partial_evidence_for_feature = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#partial_evidence_for_feature" );
    
    public static final ObjectProperty position_of = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#position_of" );
    
    /** <p>Example: miRNA processed_from miRNA_primary_transcript.</p> */
    public static final ObjectProperty processed_from = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#processed_from" );
    
    /** <p>Example: miRNA_primary_transcript processed into miRNA.</p> */
    public static final ObjectProperty processed_into = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#processed_into" );
    
    public static final ObjectProperty recombined_from = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#recombined_from" );
    
    public static final ObjectProperty recombined_to = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#recombined_to" );
    
    public static final ObjectProperty sequence_of = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#sequence_of" );
    
    public static final ObjectProperty similar_to = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#similar_to" );
    
    /** <p>Example: CDS started_by start_codon.</p> */
    public static final ObjectProperty started_by = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#started_by" );
    
    /** <p>Example: start_codon starts CDS.</p> */
    public static final ObjectProperty starts = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#starts" );
    
    public static final ObjectProperty trans_spliced_from = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#trans_spliced_from" );
    
    public static final ObjectProperty trans_spliced_to = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#trans_spliced_to" );
    
    /** <p>Example: primary_transcript transcribed_from gene.</p> */
    public static final ObjectProperty transcribed_from = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#transcribed_from" );
    
    /** <p>Example: gene transcribed_to primary_transcript.</p> */
    public static final ObjectProperty transcribed_to = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#transcribed_to" );
    
    /** <p>Example: codon translates_to amino_acid.</p> */
    public static final ObjectProperty translates_to = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#translates_to" );
    
    /** <p>Example: Polypeptide translation_of CDS.</p> */
    public static final ObjectProperty translation_of = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#translation_of" );
    
    /** <p>Added to SO during the immunology workshop, June 2007. This relationship was 
     *  approved by Barry Smith.</p>
     */
    public static final ObjectProperty variant_of = m_model.createObjectProperty( "http://purl.obolibrary.org/obo/so-xp.obo#variant_of" );
    
    public static final AnnotationProperty contains_PROP = m_model.createAnnotationProperty( "http://data.bioontology.org/metadata/obo/contains" );
    
    public static final AnnotationProperty part_of_PROP = m_model.createAnnotationProperty( "http://data.bioontology.org/metadata/obo/part_of" );
    
    public static final AnnotationProperty treeView = m_model.createAnnotationProperty( "http://data.bioontology.org/metadata/treeView" );
    
    public static final AnnotationProperty definition = m_model.createAnnotationProperty( "http://purl.obolibrary.org/obo/IAO_0000115" );
    
    public static final AnnotationProperty term_replaced_by = m_model.createAnnotationProperty( "http://purl.obolibrary.org/obo/IAO_0100001" );
    
    public static final AnnotationProperty amino_acid_modification = m_model.createAnnotationProperty( "http://purl.obolibrary.org/obo/so-xp.obo#AAMOD" );
    
    public static final AnnotationProperty biosapiens = m_model.createAnnotationProperty( "http://purl.obolibrary.org/obo/so-xp.obo#BS" );
    
    /** <p>database of genomic structural variation</p> */
    public static final AnnotationProperty DBVAR = m_model.createAnnotationProperty( "http://purl.obolibrary.org/obo/so-xp.obo#DBVAR" );
    
    public static final AnnotationProperty RNA_modification = m_model.createAnnotationProperty( "http://purl.obolibrary.org/obo/so-xp.obo#RNAMOD" );
    
    /** <p>SO feature annotation</p> */
    public static final AnnotationProperty SOFA = m_model.createAnnotationProperty( "http://purl.obolibrary.org/obo/so-xp.obo#SOFA" );
    
    public static final AnnotationProperty variant_annotation_term = m_model.createAnnotationProperty( "http://purl.obolibrary.org/obo/so-xp.obo#VAR" );
    
    public static final AnnotationProperty amino_acid_1_letter_code = m_model.createAnnotationProperty( "http://purl.obolibrary.org/obo/so-xp.obo#aa1" );
    
    public static final AnnotationProperty amino_acid_3_letter_code = m_model.createAnnotationProperty( "http://purl.obolibrary.org/obo/so-xp.obo#aa3" );
    
    /** <p>biosapiens protein feature ontology</p> */
    public static final AnnotationProperty biosapiens_PROP = m_model.createAnnotationProperty( "http://purl.obolibrary.org/obo/so-xp.obo#biosapiens" );
    
    public static final AnnotationProperty dbsnp_variant_terms = m_model.createAnnotationProperty( "http://purl.obolibrary.org/obo/so-xp.obo#dbsnp" );
    
    public static final AnnotationProperty DBVAR_PROP = m_model.createAnnotationProperty( "http://purl.obolibrary.org/obo/so-xp.obo#dbvar" );
    
    public static final AnnotationProperty ensembl_variant_terms = m_model.createAnnotationProperty( "http://purl.obolibrary.org/obo/so-xp.obo#ebi_variants" );
    
    public static final AnnotationProperty subset_property = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#SubsetProperty" );
    
    public static final AnnotationProperty synonym_type_property = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#SynonymTypeProperty" );
    
    public static final AnnotationProperty auto_generated_by = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#auto-generated-by" );
    
    public static final AnnotationProperty consider = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#consider" );
    
    public static final AnnotationProperty created_by = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#created_by" );
    
    public static final AnnotationProperty creation_date = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#creation_date" );
    
    public static final AnnotationProperty date = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#date" );
    
    public static final AnnotationProperty default_namespace = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#default-namespace" );
    
    public static final AnnotationProperty has_alternative_id = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#hasAlternativeId" );
    
    public static final AnnotationProperty has_broad_synonym = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#hasBroadSynonym" );
    
    public static final AnnotationProperty database_cross_reference = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#hasDbXref" );
    
    public static final AnnotationProperty has_exact_synonym = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#hasExactSynonym" );
    
    public static final AnnotationProperty has_narrow_synonym = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#hasNarrowSynonym" );
    
    public static final AnnotationProperty has_obo_format_version = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#hasOBOFormatVersion" );
    
    public static final AnnotationProperty has_obo_namespace = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#hasOBONamespace" );
    
    public static final AnnotationProperty has_related_synonym = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#hasRelatedSynonym" );
    
    public static final AnnotationProperty has_scope = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#hasScope" );
    
    public static final AnnotationProperty has_synonym_type = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#hasSynonymType" );
    
    public static final AnnotationProperty id = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#id" );
    
    public static final AnnotationProperty in_subset = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#inSubset" );
    
    public static final AnnotationProperty remark = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#remark" );
    
    public static final AnnotationProperty saved_by = m_model.createAnnotationProperty( "http://www.geneontology.org/formats/oboInOwl#saved-by" );
    
    public static final AnnotationProperty comment = m_model.createAnnotationProperty( "http://www.w3.org/2000/01/rdf-schema#comment" );
    
    public static final AnnotationProperty label = m_model.createAnnotationProperty( "http://www.w3.org/2000/01/rdf-schema#label" );
    
    public static final AnnotationProperty deprecated = m_model.createAnnotationProperty( "http://www.w3.org/2002/07/owl#deprecated" );
    
    public static final AnnotationProperty notation = m_model.createAnnotationProperty( "http://www.w3.org/2004/02/skos/core#notation" );
    
    public static final OntClass Sequence_Ontology = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000000" );
    
    public static final OntClass region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000001" );
    
    public static final OntClass sequence_secondary_structure = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000002" );
    
    public static final OntClass G_quartet = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000003" );
    
    public static final OntClass interior_coding_exon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000004" );
    
    public static final OntClass satellite_DNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000005" );
    
    /** <p>This term is mapped to MGED. This term is now located in OBI, with the following 
     *  ID OBI_0000406.</p>
     */
    public static final OntClass PCR_product = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000006" );
    
    public static final OntClass read_pair = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000007" );
    
    public static final OntClass gene_sensu_your_favorite_organism = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000008" );
    
    public static final OntClass gene_class = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000009" );
    
    public static final OntClass protein_coding = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000010" );
    
    public static final OntClass non_protein_coding = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000011" );
    
    public static final OntClass scRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000012" );
    
    public static final OntClass scRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000013" );
    
    /** <p>Binds TAF1, TAF2.</p> */
    public static final OntClass INR_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000014" );
    
    /** <p>Binds TAF6, TAF9.</p> */
    public static final OntClass DPE_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000015" );
    
    /** <p>Binds TFIIB.</p> */
    public static final OntClass BREu_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000016" );
    
    public static final OntClass PSE_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000017" );
    
    public static final OntClass linkage_group = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000018" );
    
    public static final OntClass RNA_internal_loop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000020" );
    
    public static final OntClass asymmetric_RNA_internal_loop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000021" );
    
    public static final OntClass A_minor_RNA_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000022" );
    
    public static final OntClass K_turn_RNA_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000023" );
    
    public static final OntClass sarcin_like_RNA_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000024" );
    
    public static final OntClass symmetric_RNA_internal_loop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000025" );
    
    public static final OntClass RNA_junction_loop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000026" );
    
    public static final OntClass RNA_hook_turn = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000027" );
    
    public static final OntClass base_pair = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000028" );
    
    public static final OntClass WC_base_pair = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000029" );
    
    public static final OntClass sugar_edge_base_pair = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000030" );
    
    public static final OntClass aptamer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000031" );
    
    public static final OntClass DNA_aptamer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000032" );
    
    public static final OntClass RNA_aptamer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000033" );
    
    public static final OntClass morpholino_oligo = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000034" );
    
    public static final OntClass riboswitch = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000035" );
    
    public static final OntClass matrix_attachment_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000036" );
    
    public static final OntClass locus_control_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000037" );
    
    public static final OntClass match_set = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000038" );
    
    public static final OntClass match_part = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000039" );
    
    public static final OntClass genomic_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000040" );
    
    public static final OntClass sequence_operation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000041" );
    
    public static final OntClass pseudogene_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000042" );
    
    /** <p>Please not the synonym R psi M uses the spelled out form of the greek letter.</p> */
    public static final OntClass processed_pseudogene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000043" );
    
    public static final OntClass pseudogene_by_unequal_crossing_over = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000044" );
    
    public static final OntClass delete = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000045" );
    
    public static final OntClass insert = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000046" );
    
    public static final OntClass invert = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000047" );
    
    public static final OntClass substitute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000048" );
    
    public static final OntClass translocate = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000049" );
    
    public static final OntClass gene_part = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000050" );
    
    public static final OntClass probe = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000051" );
    
    public static final OntClass assortment_derived_deficiency = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000052" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_regulatory_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000053" );
    
    public static final OntClass aneuploid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000054" );
    
    public static final OntClass hyperploid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000055" );
    
    public static final OntClass hypoploid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000056" );
    
    public static final OntClass operator = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000057" );
    
    public static final OntClass assortment_derived_aneuploid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000058" );
    
    public static final OntClass nuclease_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000059" );
    
    /** <p>FLAG - this term is should probably be a part of rather than an is_a.</p> */
    public static final OntClass compound_chromosome_arm = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000060" );
    
    /** <p>A region of a molecule that binds to a restriction enzyme.</p> */
    public static final OntClass restriction_enzyme_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000061" );
    
    public static final OntClass deficient_intrachromosomal_transposition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000062" );
    
    public static final OntClass deficient_interchromosomal_transposition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000063" );
    
    /** <p>This classes of attributes was added by MA to allow the broad description 
     *  of genes based on qualities of the transcript(s). A product of SO meeting 
     *  2004.</p>
     */
    public static final OntClass gene_by_transcript_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000064" );
    
    public static final OntClass free_chromosome_arm = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000065" );
    
    public static final OntClass gene_by_polyadenylation_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000066" );
    
    public static final OntClass gene_to_gene_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000067" );
    
    public static final OntClass overlapping = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000068" );
    
    public static final OntClass inside_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000069" );
    
    public static final OntClass inside_intron_antiparallel = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000070" );
    
    public static final OntClass inside_intron_parallel = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000071" );
    
    public static final OntClass end_overlapping_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000072" );
    
    public static final OntClass five_prime_three_prime_overlap = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000073" );
    
    public static final OntClass five_prime_five_prime_overlap = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000074" );
    
    public static final OntClass three_prime_three_prime_overlap = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000075" );
    
    public static final OntClass three_prime_five_prime_overlap = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000076" );
    
    public static final OntClass antisense = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000077" );
    
    public static final OntClass polycistronic_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000078" );
    
    public static final OntClass dicistronic_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000079" );
    
    public static final OntClass operon_member = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000080" );
    
    public static final OntClass gene_array_member = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000081" );
    
    public static final OntClass processed_transcript_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000082" );
    
    public static final OntClass macronuclear_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000083" );
    
    public static final OntClass micronuclear_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000084" );
    
    public static final OntClass gene_by_genome_location = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000085" );
    
    public static final OntClass gene_by_organelle_of_genome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000086" );
    
    public static final OntClass nuclear_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000087" );
    
    public static final OntClass mt_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000088" );
    
    public static final OntClass kinetoplast_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000089" );
    
    public static final OntClass plastid_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000090" );
    
    public static final OntClass apicoplast_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000091" );
    
    public static final OntClass ct_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000092" );
    
    public static final OntClass chromoplast_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000093" );
    
    public static final OntClass cyanelle_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000094" );
    
    public static final OntClass leucoplast_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000095" );
    
    public static final OntClass proplastid_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000096" );
    
    public static final OntClass nucleomorph_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000097" );
    
    public static final OntClass plasmid_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000098" );
    
    public static final OntClass proviral_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000099" );
    
    public static final OntClass endogenous_retroviral_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000100" );
    
    public static final OntClass transposable_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000101" );
    
    public static final OntClass expressed_sequence_match = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000102" );
    
    public static final OntClass clone_insert_end = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000103" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology. 
     *  The term 'protein' was merged with 'polypeptide'. Although 'protein' was a 
     *  sequence_attribute and therefore meant to describe the quality rather than 
     *  an actual feature, it was being used erroneously. It is replaced by 'peptidyl' 
     *  as the polymer attribute.</p>
     */
    public static final OntClass polypeptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000104" );
    
    public static final OntClass chromosome_arm = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000105" );
    
    public static final OntClass non_capped_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000106" );
    
    public static final OntClass sequencing_primer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000107" );
    
    public static final OntClass mRNA_with_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000108" );
    
    public static final OntClass sequence_variant_obs = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000109" );
    
    public static final OntClass sequence_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000110" );
    
    public static final OntClass transposable_element_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000111" );
    
    public static final OntClass primer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000112" );
    
    public static final OntClass proviral_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000113" );
    
    public static final OntClass methylated_cytosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000114" );
    
    public static final OntClass transcript_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000115" );
    
    public static final OntClass edited = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000116" );
    
    public static final OntClass transcript_with_readthrough_stop_codon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000117" );
    
    public static final OntClass transcript_with_translational_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000118" );
    
    public static final OntClass regulated = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000119" );
    
    /** <p>May contain introns.</p> */
    public static final OntClass protein_coding_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000120" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass forward_primer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000121" );
    
    public static final OntClass RNA_sequence_secondary_structure = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000122" );
    
    /** <p>By:&lt;protein_id&gt;.</p> */
    public static final OntClass transcriptionally_regulated = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000123" );
    
    public static final OntClass transcriptionally_constitutive = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000124" );
    
    public static final OntClass transcriptionally_induced = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000125" );
    
    public static final OntClass transcriptionally_repressed = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000126" );
    
    public static final OntClass silenced_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000127" );
    
    public static final OntClass gene_silenced_by_DNA_modification = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000128" );
    
    public static final OntClass gene_silenced_by_DNA_methylation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000129" );
    
    public static final OntClass post_translationally_regulated = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000130" );
    
    public static final OntClass translationally_regulated = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000131" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass reverse_primer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000132" );
    
    public static final OntClass epigenetically_modified = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000133" );
    
    public static final OntClass genomically_imprinted = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000134" );
    
    public static final OntClass maternally_imprinted = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000135" );
    
    public static final OntClass paternally_imprinted = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000136" );
    
    /** <p>Examples are x-inactivation and immunoglobulin formation.</p> */
    public static final OntClass allelically_excluded = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000137" );
    
    public static final OntClass gene_rearranged_at_DNA_level = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000138" );
    
    public static final OntClass ribosome_entry_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000139" );
    
    public static final OntClass attenuator = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000140" );
    
    public static final OntClass terminator = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000141" );
    
    public static final OntClass DNA_sequence_secondary_structure = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000142" );
    
    public static final OntClass assembly_component = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000143" );
    
    public static final OntClass primary_transcript_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000144" );
    
    public static final OntClass recoded_codon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000145" );
    
    public static final OntClass capped = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000146" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass exon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000147" );
    
    public static final OntClass supercontig = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000148" );
    
    public static final OntClass contig = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000149" );
    
    public static final OntClass read = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000150" );
    
    public static final OntClass clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000151" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass YAC = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000152" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass BAC = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000153" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology. 
     *  Drosophila melanogaster PACs carry an average insert size of 80 kb. The library 
     *  represents a 6-fold coverage of the genome.</p>
     */
    public static final OntClass PAC = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000154" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass plasmid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000155" );
    
    /** <p>Paper: vans GA et al. High efficiency vectors for cosmid microcloning and 
     *  genomic analysis. Gene 1989; 79(1):9-20. This term is mapped to MGED. Do not 
     *  obsolete without consulting MGED ontology.</p>
     */
    public static final OntClass cosmid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000156" );
    
    public static final OntClass phagemid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000157" );
    
    /** <p>Birren BW et al. A human chromosome 22 fosmid resource: mapping and analysis 
     *  of 96 clones. Genomics 1996.</p>
     */
    public static final OntClass fosmid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000158" );
    
    public static final OntClass deletion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000159" );
    
    public static final OntClass lambda_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000160" );
    
    public static final OntClass methylated_adenine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000161" );
    
    /** <p>With spliceosomal introns, the splice sites bind the spliceosomal machinery.</p> */
    public static final OntClass splice_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000162" );
    
    public static final OntClass five_prime_cis_splice_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000163" );
    
    public static final OntClass three_prime_cis_splice_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000164" );
    
    /** <p>An enhancer may participate in an enhanceosome GO:0034206. A protein-DNA complex 
     *  formed by the association of a distinct set of general and specific transcription 
     *  factors with a region of enhancer DNA. The cooperative assembly of an enhanceosome 
     *  confers specificity of transcriptional regulation. This comment is a place 
     *  holder should we start to make cross products with GO.</p>
     */
    public static final OntClass enhancer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000165" );
    
    public static final OntClass enhancer_bound_by_factor = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000166" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology. 
     *  The region on a DNA molecule involved in RNA polymerase binding to initiate 
     *  transcription.</p>
     */
    public static final OntClass promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000167" );
    
    public static final OntClass restriction_enzyme_cut_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000168" );
    
    public static final OntClass RNApol_I_promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000169" );
    
    public static final OntClass RNApol_II_promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000170" );
    
    public static final OntClass RNApol_III_promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000171" );
    
    public static final OntClass CAAT_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000172" );
    
    public static final OntClass GC_rich_promoter_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000173" );
    
    /** <p>Binds TBP.</p> */
    public static final OntClass TATA_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000174" );
    
    public static final OntClass minus_10_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000175" );
    
    public static final OntClass minus_35_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000176" );
    
    public static final OntClass cross_genome_match = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000177" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass operon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000178" );
    
    public static final OntClass clone_insert_start = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000179" );
    
    public static final OntClass retrotransposon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000180" );
    
    public static final OntClass translated_nucleotide_match = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000181" );
    
    public static final OntClass DNA_transposon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000182" );
    
    public static final OntClass non_transcribed_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000183" );
    
    /** <p>May have either GT-AG or AT-AG 5' and 3' boundaries.</p> */
    public static final OntClass U2_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000184" );
    
    public static final OntClass primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000185" );
    
    public static final OntClass LTR_retrotransposon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000186" );
    
    public static final OntClass repeat_family = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000187" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000188" );
    
    public static final OntClass non_LTR_retrotransposon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000189" );
    
    public static final OntClass five_prime_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000190" );
    
    public static final OntClass interior_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000191" );
    
    public static final OntClass three_prime_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000192" );
    
    public static final OntClass RFLP_fragment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000193" );
    
    public static final OntClass LINE_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000194" );
    
    public static final OntClass coding_exon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000195" );
    
    public static final OntClass five_prime_coding_exon_coding_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000196" );
    
    public static final OntClass three_prime_coding_exon_coding_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000197" );
    
    public static final OntClass noncoding_exon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000198" );
    
    public static final OntClass translocation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000199" );
    
    public static final OntClass five_prime_coding_exon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000200" );
    
    public static final OntClass interior_exon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000201" );
    
    public static final OntClass three_prime_coding_exon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000202" );
    
    public static final OntClass UTR = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000203" );
    
    public static final OntClass five_prime_UTR = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000204" );
    
    public static final OntClass three_prime_UTR = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000205" );
    
    public static final OntClass SINE_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000206" );
    
    public static final OntClass simple_sequence_length_variation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000207" );
    
    public static final OntClass terminal_inverted_repeat_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000208" );
    
    public static final OntClass rRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000209" );
    
    public static final OntClass tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000210" );
    
    public static final OntClass alanine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000211" );
    
    public static final OntClass arginine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000212" );
    
    public static final OntClass asparagine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000213" );
    
    public static final OntClass aspartic_acid_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000214" );
    
    public static final OntClass cysteine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000215" );
    
    public static final OntClass glutamic_acid_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000216" );
    
    public static final OntClass glutamine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000217" );
    
    public static final OntClass glycine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000218" );
    
    public static final OntClass histidine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000219" );
    
    public static final OntClass isoleucine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000220" );
    
    public static final OntClass leucine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000221" );
    
    public static final OntClass lysine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000222" );
    
    public static final OntClass methionine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000223" );
    
    public static final OntClass phenylalanine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000224" );
    
    public static final OntClass proline_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000225" );
    
    public static final OntClass serine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000226" );
    
    public static final OntClass threonine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000227" );
    
    public static final OntClass tryptophan_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000228" );
    
    public static final OntClass tyrosine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000229" );
    
    public static final OntClass valine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000230" );
    
    public static final OntClass snRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000231" );
    
    public static final OntClass snoRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000232" );
    
    /** <p>A processed transcript cannot contain introns.</p> */
    public static final OntClass mature_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000233" );
    
    /** <p>An mRNA does not contain introns as it is a processed_transcript. The equivalent 
     *  kind of primary_transcript is protein_coding_primary_transcript (SO:0000120) 
     *  which may contain introns. This term is mapped to MGED. Do not obsolete without 
     *  consulting MGED ontology.</p>
     */
    public static final OntClass mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000234" );
    
    public static final OntClass TF_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000235" );
    
    /** <p>The definition was modified by Rama. ORF is defined by the sequence, whereas 
     *  the CDS is defined according to whether a polypeptide is made. This term is 
     *  mapped to MGED. Do not obsolete without consulting MGED ontology.</p>
     */
    public static final OntClass ORF = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000236" );
    
    public static final OntClass transcript_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000237" );
    
    public static final OntClass foldback_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000238" );
    
    public static final OntClass flanking_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000239" );
    
    public static final OntClass chromosome_variation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000240" );
    
    public static final OntClass internal_UTR = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000241" );
    
    public static final OntClass untranslated_region_polycistronic_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000242" );
    
    public static final OntClass internal_ribosome_entry_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000243" );
    
    public static final OntClass four_cutter_restriction_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000244" );
    
    public static final OntClass mRNA_by_polyadenylation_status = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000245" );
    
    public static final OntClass polyadenylated = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000246" );
    
    public static final OntClass mRNA_not_polyadenylated = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000247" );
    
    public static final OntClass sequence_length_variation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000248" );
    
    public static final OntClass six_cutter_restriction_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000249" );
    
    public static final OntClass modified_RNA_base_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000250" );
    
    public static final OntClass eight_cutter_restriction_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000251" );
    
    public static final OntClass rRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000252" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000253" );
    
    public static final OntClass alanyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000254" );
    
    public static final OntClass rRNA_small_subunit_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000255" );
    
    public static final OntClass asparaginyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000256" );
    
    public static final OntClass aspartyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000257" );
    
    public static final OntClass cysteinyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000258" );
    
    public static final OntClass glutaminyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000259" );
    
    public static final OntClass glutamyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000260" );
    
    public static final OntClass glycyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000261" );
    
    public static final OntClass histidyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000262" );
    
    public static final OntClass isoleucyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000263" );
    
    public static final OntClass leucyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000264" );
    
    public static final OntClass lysyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000265" );
    
    public static final OntClass methionyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000266" );
    
    public static final OntClass phenylalanyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000267" );
    
    public static final OntClass prolyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000268" );
    
    public static final OntClass seryl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000269" );
    
    public static final OntClass threonyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000270" );
    
    public static final OntClass tryptophanyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000271" );
    
    public static final OntClass tyrosyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000272" );
    
    public static final OntClass valyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000273" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass snRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000274" );
    
    public static final OntClass snoRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000275" );
    
    public static final OntClass miRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000276" );
    
    /** <p>Formerly called transcript_by_bound_factor.</p> */
    public static final OntClass bound_by_factor = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000277" );
    
    /** <p>Formerly called transcript_by_bound_nucleic_acid.</p> */
    public static final OntClass transcript_bound_by_nucleic_acid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000278" );
    
    /** <p>Formerly called transcript_by_bound_protein.</p> */
    public static final OntClass transcript_bound_by_protein = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000279" );
    
    public static final OntClass engineered_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000280" );
    
    public static final OntClass engineered_foreign_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000281" );
    
    public static final OntClass mRNA_with_minus_1_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000282" );
    
    public static final OntClass engineered_foreign_transposable_element_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000283" );
    
    public static final OntClass type_I_enzyme_restriction_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000284" );
    
    public static final OntClass foreign_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000285" );
    
    public static final OntClass long_terminal_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000286" );
    
    public static final OntClass fusion_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000287" );
    
    public static final OntClass engineered_fusion_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000288" );
    
    public static final OntClass microsatellite = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000289" );
    
    public static final OntClass dinucleotide_repeat_microsatellite_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000290" );
    
    public static final OntClass trinucleotide_repeat_microsatellite_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000291" );
    
    public static final OntClass repetitive_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000292" );
    
    public static final OntClass engineered_foreign_repetitive_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000293" );
    
    public static final OntClass inverted_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000294" );
    
    /** <p>May have either GT-AC or AT-AC 5' and 3' boundaries.</p> */
    public static final OntClass U12_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000295" );
    
    public static final OntClass origin_of_replication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000296" );
    
    public static final OntClass D_loop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000297" );
    
    public static final OntClass recombination_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000298" );
    
    public static final OntClass specific_recombination_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000299" );
    
    public static final OntClass recombination_feature_of_rearranged_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000300" );
    
    public static final OntClass vertebrate_immune_system_gene_recombination_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000301" );
    
    public static final OntClass J_gene_recombination_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000302" );
    
    public static final OntClass clip = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000303" );
    
    public static final OntClass type_II_enzyme_restriction_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000304" );
    
    /** <p>Modified base:&lt;modified_base&gt;.</p> */
    public static final OntClass modified_DNA_base = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000305" );
    
    public static final OntClass methylated_DNA_base_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000306" );
    
    public static final OntClass CpG_island = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000307" );
    
    public static final OntClass sequence_feature_locating_method = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000308" );
    
    public static final OntClass computed_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000309" );
    
    public static final OntClass predicted_ab_initio_computation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000310" );
    
    /** <p>similar to:&lt;sequence_id&gt;</p> */
    public static final OntClass computed_feature_by_similarity = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000311" );
    
    public static final OntClass experimentally_determined = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000312" );
    
    public static final OntClass stem_loop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000313" );
    
    public static final OntClass direct_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000314" );
    
    public static final OntClass TSS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000315" );
    
    public static final OntClass CDS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000316" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass cDNA_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000317" );
    
    public static final OntClass start_codon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000318" );
    
    public static final OntClass stop_codon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000319" );
    
    public static final OntClass intronic_splice_enhancer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000320" );
    
    public static final OntClass mRNA_with_plus_1_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000321" );
    
    public static final OntClass nuclease_hypersensitive_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000322" );
    
    public static final OntClass coding_start = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000323" );
    
    public static final OntClass tag = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000324" );
    
    public static final OntClass rRNA_large_subunit_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000325" );
    
    public static final OntClass SAGE_tag = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000326" );
    
    public static final OntClass coding_end = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000327" );
    
    public static final OntClass microarray_oligo = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000328" );
    
    public static final OntClass mRNA_with_plus_2_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000329" );
    
    public static final OntClass conserved_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000330" );
    
    public static final OntClass STS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000331" );
    
    public static final OntClass coding_conserved_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000332" );
    
    public static final OntClass exon_junction = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000333" );
    
    public static final OntClass nc_conserved_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000334" );
    
    public static final OntClass mRNA_with_minus_2_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000335" );
    
    public static final OntClass pseudogene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000336" );
    
    public static final OntClass RNAi_reagent = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000337" );
    
    public static final OntClass MITE = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000338" );
    
    public static final OntClass recombination_hotspot = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000339" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000340" );
    
    public static final OntClass chromosome_band = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000341" );
    
    public static final OntClass site_specific_recombination_target_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000342" );
    
    public static final OntClass match = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000343" );
    
    public static final OntClass splice_enhancer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000344" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass EST = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000345" );
    
    public static final OntClass loxP_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000346" );
    
    public static final OntClass nucleotide_match = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000347" );
    
    public static final OntClass nucleic_acid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000348" );
    
    public static final OntClass protein_match = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000349" );
    
    public static final OntClass FRT_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000350" );
    
    public static final OntClass synthetic_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000351" );
    
    public static final OntClass DNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000352" );
    
    public static final OntClass sequence_assembly = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000353" );
    
    public static final OntClass group_1_intron_homing_endonuclease_target_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000354" );
    
    public static final OntClass haplotype_block = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000355" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000356" );
    
    public static final OntClass flanked = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000357" );
    
    public static final OntClass floxed = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000359" );
    
    public static final OntClass codon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000360" );
    
    public static final OntClass FRT_flanked = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000361" );
    
    public static final OntClass invalidated_by_chimeric_cDNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000362" );
    
    public static final OntClass floxed_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000363" );
    
    public static final OntClass transposable_element_flanking_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000364" );
    
    public static final OntClass integron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000365" );
    
    public static final OntClass insertion_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000366" );
    
    public static final OntClass attI_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000367" );
    
    public static final OntClass transposable_element_insertion_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000368" );
    
    public static final OntClass integrase_coding_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000369" );
    
    public static final OntClass small_regulatory_ncRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000370" );
    
    public static final OntClass conjugative_transposon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000371" );
    
    /** <p>This was moved to be a child of transcript (SO:0000673) because some enzymatic 
     *  RNA regions are part of primary transcripts and some are part of processed 
     *  transcripts.</p>
     */
    public static final OntClass enzymatic_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000372" );
    
    public static final OntClass recombinationally_inverted_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000373" );
    
    public static final OntClass ribozyme = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000374" );
    
    public static final OntClass rRNA_5_8S = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000375" );
    
    public static final OntClass RNA_6S = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000376" );
    
    public static final OntClass CsrB_RsmB_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000377" );
    
    public static final OntClass DsrA_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000378" );
    
    public static final OntClass GcvB_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000379" );
    
    public static final OntClass hammerhead_ribozyme = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000380" );
    
    public static final OntClass group_IIA_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000381" );
    
    public static final OntClass group_IIB_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000382" );
    
    public static final OntClass MicF_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000383" );
    
    public static final OntClass OxyS_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000384" );
    
    public static final OntClass RNase_MRP_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000385" );
    
    public static final OntClass RNase_P_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000386" );
    
    public static final OntClass RprA_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000387" );
    
    public static final OntClass RRE_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000388" );
    
    public static final OntClass spot_42_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000389" );
    
    public static final OntClass telomerase_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000390" );
    
    public static final OntClass U1_snRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000391" );
    
    public static final OntClass U2_snRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000392" );
    
    public static final OntClass U4_snRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000393" );
    
    public static final OntClass U4atac_snRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000394" );
    
    public static final OntClass U5_snRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000395" );
    
    public static final OntClass U6_snRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000396" );
    
    public static final OntClass U6atac_snRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000397" );
    
    public static final OntClass U11_snRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000398" );
    
    public static final OntClass U12_snRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000399" );
    
    public static final OntClass sequence_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000400" );
    
    public static final OntClass gene_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000401" );
    
    public static final OntClass enhancer_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000402" );
    
    /** <p>An evolutionarily conserved eukaryotic low molecular weight RNA capable of 
     *  intermolecular hybridization with both homologous and heterologous 18S rRNA.</p>
     */
    public static final OntClass U14_snoRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000403" );
    
    public static final OntClass vault_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000404" );
    
    public static final OntClass Y_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000405" );
    
    public static final OntClass twintron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000406" );
    
    public static final OntClass rRNA_18S = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000407" );
    
    public static final OntClass site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000408" );
    
    /** <p>See GO:0005488 : binding.</p> */
    public static final OntClass binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000409" );
    
    /** <p>See GO:0042277 : peptide binding.</p> */
    public static final OntClass protein_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000410" );
    
    public static final OntClass rescue_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000411" );
    
    public static final OntClass restriction_fragment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000412" );
    
    public static final OntClass sequence_difference = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000413" );
    
    public static final OntClass invalidated_by_genomic_contamination = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000414" );
    
    public static final OntClass invalidated_by_genomic_polyA_primed_cDNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000415" );
    
    public static final OntClass invalidated_by_partial_processing = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000416" );
    
    /** <p>Range. Old definition from before biosapiens: A region of a single polypeptide 
     *  chain that folds into an independent unit and exhibits biological activity. 
     *  A polypeptide chain may have multiple domains.</p>
     */
    public static final OntClass polypeptide_domain = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000417" );
    
    /** <p>Old def before biosapiens:The sequence for an N-terminal domain of a secreted 
     *  protein; this domain is involved in attaching nascent polypeptide to the membrane 
     *  leader sequence.</p>
     */
    public static final OntClass signal_peptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000418" );
    
    /** <p>This term mature peptide, merged with the biosapiens term mature protein region 
     *  and took that to be the new name. Old def: The coding sequence for the mature 
     *  or final peptide or protein product following post-translational modification.</p>
     */
    public static final OntClass mature_protein_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000419" );
    
    public static final OntClass five_prime_terminal_inverted_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000420" );
    
    public static final OntClass three_prime_terminal_inverted_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000421" );
    
    public static final OntClass U5_LTR_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000422" );
    
    public static final OntClass R_LTR_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000423" );
    
    public static final OntClass U3_LTR_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000424" );
    
    public static final OntClass five_prime_LTR = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000425" );
    
    public static final OntClass three_prime_LTR = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000426" );
    
    public static final OntClass R_five_prime_LTR_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000427" );
    
    public static final OntClass U5_five_prime_LTR_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000428" );
    
    public static final OntClass U3_five_prime_LTR_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000429" );
    
    public static final OntClass R_three_prime_LTR_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000430" );
    
    public static final OntClass U3_three_prime_LTR_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000431" );
    
    public static final OntClass U5_three_prime_LTR_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000432" );
    
    public static final OntClass non_LTR_retrotransposon_polymeric_tract = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000433" );
    
    public static final OntClass target_site_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000434" );
    
    public static final OntClass RR_tract = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000435" );
    
    public static final OntClass ARS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000436" );
    
    public static final OntClass assortment_derived_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000437" );
    
    public static final OntClass gene_not_polyadenylated = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000438" );
    
    public static final OntClass inverted_ring_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000439" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass vector_replicon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000440" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass ss_oligo = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000441" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass ds_oligo = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000442" );
    
    public static final OntClass polymer_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000443" );
    
    public static final OntClass three_prime_noncoding_exon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000444" );
    
    public static final OntClass five_prime_noncoding_exon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000445" );
    
    public static final OntClass UTR_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000446" );
    
    public static final OntClass five_prime_UTR_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000447" );
    
    public static final OntClass three_prime_UTR_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000448" );
    
    public static final OntClass random_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000449" );
    
    public static final OntClass interband = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000450" );
    
    public static final OntClass gene_with_polyadenylated_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000451" );
    
    public static final OntClass transgene_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000452" );
    
    public static final OntClass chromosomal_transposition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000453" );
    
    public static final OntClass rasiRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000454" );
    
    public static final OntClass gene_with_mRNA_with_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000455" );
    
    public static final OntClass recombinationally_rearranged_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000456" );
    
    public static final OntClass interchromosomal_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000457" );
    
    public static final OntClass D_gene_segment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000458" );
    
    public static final OntClass gene_with_trans_spliced_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000459" );
    
    /** <p>I am using the term segment instead of gene here to avoid confusion with the 
     *  region 'gene'.</p>
     */
    public static final OntClass vertebrate_immunoglobulin_T_cell_receptor_segment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000460" );
    
    public static final OntClass inversion_derived_bipartite_deficiency = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000461" );
    
    public static final OntClass pseudogenic_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000462" );
    
    public static final OntClass encodes_alternately_spliced_transcripts = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000463" );
    
    public static final OntClass decayed_exon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000464" );
    
    public static final OntClass inversion_derived_deficiency_plus_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000465" );
    
    public static final OntClass V_gene_segment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000466" );
    
    public static final OntClass post_translationally_regulated_by_protein_stability = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000467" );
    
    public static final OntClass golden_path_fragment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000468" );
    
    public static final OntClass post_translationally_regulated_by_protein_modification = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000469" );
    
    public static final OntClass J_gene_segment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000470" );
    
    public static final OntClass autoregulated = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000471" );
    
    public static final OntClass tiling_path = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000472" );
    
    public static final OntClass negatively_autoregulated = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000473" );
    
    public static final OntClass tiling_path_fragment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000474" );
    
    public static final OntClass positively_autoregulated = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000475" );
    
    public static final OntClass contig_read = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000476" );
    
    public static final OntClass polycistronic_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000477" );
    
    public static final OntClass C_gene_segment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000478" );
    
    public static final OntClass trans_spliced_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000479" );
    
    public static final OntClass tiling_path_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000480" );
    
    public static final OntClass terminal_inverted_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000481" );
    
    public static final OntClass vertebrate_immunoglobulin_T_cell_receptor_gene_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000482" );
    
    public static final OntClass nc_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000483" );
    
    public static final OntClass three_prime_coding_exon_noncoding_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000484" );
    
    public static final OntClass DJ_J_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000485" );
    
    public static final OntClass five_prime_coding_exon_noncoding_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000486" );
    
    public static final OntClass VDJ_J_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000487" );
    
    public static final OntClass VDJ_J_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000488" );
    
    public static final OntClass VJ_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000489" );
    
    public static final OntClass VJ_J_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000490" );
    
    public static final OntClass VJ_J_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000491" );
    
    public static final OntClass D_gene_recombination_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000492" );
    
    public static final OntClass three_prime_D_heptamer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000493" );
    
    public static final OntClass three_prime_D_nonamer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000494" );
    
    public static final OntClass three_prime_D_spacer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000495" );
    
    public static final OntClass five_prime_D_heptamer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000496" );
    
    public static final OntClass five_prime_D_nonamer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000497" );
    
    public static final OntClass five_prime_D_spacer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000498" );
    
    public static final OntClass virtual_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000499" );
    
    public static final OntClass Hoogsteen_base_pair = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000500" );
    
    public static final OntClass reverse_Hoogsteen_base_pair = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000501" );
    
    /** <p>This concept cam about as a direct result of the SO meeting August 2004.nThe 
     *  exact nature of the relationship between transcribed_region and gene is still 
     *  up for discussion. We are going with 'associated_with' for the time being.</p>
     */
    public static final OntClass transcribed_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000502" );
    
    public static final OntClass alternately_spliced_gene_encodeing_one_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000503" );
    
    public static final OntClass D_DJ_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000504" );
    
    public static final OntClass D_DJ_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000505" );
    
    public static final OntClass D_DJ_J_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000506" );
    
    /** <p>This is the analog of the exon of a functional gene. The term was requested 
     *  by Rama - SGD to allow the annotation of the parts of a pseudogene. Non-functional 
     *  is defined as either its transcription or translation (or both) are prevented 
     *  due to one or more mutations.</p>
     */
    public static final OntClass pseudogenic_exon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000507" );
    
    public static final OntClass D_DJ_J_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000508" );
    
    public static final OntClass D_J_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000509" );
    
    public static final OntClass VD_gene_segment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000510" );
    
    public static final OntClass J_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000511" );
    
    public static final OntClass inversion_derived_deficiency_plus_aneuploid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000512" );
    
    public static final OntClass J_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000513" );
    
    public static final OntClass J_nonamer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000514" );
    
    public static final OntClass J_heptamer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000515" );
    
    /** <p>This is the analog of the transcript of a functional gene. The term was requested 
     *  by Rama - SGD to allow the annotation of the parts of a pseudogene. Non-functional 
     *  is defined as either its transcription or translation (or both) are prevented 
     *  due to one or more mutations.</p>
     */
    public static final OntClass pseudogenic_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000516" );
    
    public static final OntClass J_spacer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000517" );
    
    public static final OntClass V_DJ_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000518" );
    
    public static final OntClass V_DJ_J_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000519" );
    
    public static final OntClass V_VDJ_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000520" );
    
    public static final OntClass V_VDJ_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000521" );
    
    public static final OntClass V_VDJ_J_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000522" );
    
    public static final OntClass V_VJ_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000523" );
    
    public static final OntClass V_VJ_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000524" );
    
    public static final OntClass V_VJ_J_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000525" );
    
    public static final OntClass V_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000526" );
    
    public static final OntClass V_D_DJ_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000527" );
    
    public static final OntClass V_D_DJ_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000528" );
    
    public static final OntClass V_D_DJ_J_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000529" );
    
    public static final OntClass V_D_DJ_J_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000530" );
    
    public static final OntClass V_D_J_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000531" );
    
    public static final OntClass V_D_J_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000532" );
    
    public static final OntClass V_heptamer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000533" );
    
    public static final OntClass V_J_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000534" );
    
    public static final OntClass V_J_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000535" );
    
    public static final OntClass V_nonamer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000536" );
    
    public static final OntClass V_spacer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000537" );
    
    public static final OntClass V_gene_recombination_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000538" );
    
    public static final OntClass DJ_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000539" );
    
    public static final OntClass DJ_J_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000540" );
    
    public static final OntClass VDJ_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000541" );
    
    public static final OntClass V_DJ_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000542" );
    
    public static final OntClass alternately_spliced_gene_encoding_greater_than_one_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000543" );
    
    public static final OntClass helitron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000544" );
    
    public static final OntClass recoding_pseudoknot = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000545" );
    
    public static final OntClass designed_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000546" );
    
    public static final OntClass inversion_derived_bipartite_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000547" );
    
    public static final OntClass gene_with_edited_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000548" );
    
    public static final OntClass inversion_derived_duplication_plus_aneuploid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000549" );
    
    /** <p>Examples are Nullo-4, Haplo-4 and triplo-4 in Drosophila.</p> */
    public static final OntClass aneuploid_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000550" );
    
    public static final OntClass polyA_signal_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000551" );
    
    /** <p>Not found in Eukaryotic sequence.</p> */
    public static final OntClass Shine_Dalgarno_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000552" );
    
    public static final OntClass polyA_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000553" );
    
    public static final OntClass assortment_derived_deficiency_plus_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000554" );
    
    public static final OntClass five_prime_clip = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000555" );
    
    public static final OntClass five_prime_D_recombination_signal_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000556" );
    
    public static final OntClass three_prime_clip = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000557" );
    
    public static final OntClass C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000558" );
    
    public static final OntClass D_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000559" );
    
    public static final OntClass D_J_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000560" );
    
    public static final OntClass heptamer_of_recombination_feature_of_vertebrate_immune_system_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000561" );
    
    public static final OntClass nonamer_of_recombination_feature_of_vertebrate_immune_system_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000562" );
    
    public static final OntClass vertebrate_immune_system_gene_recombination_spacer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000563" );
    
    public static final OntClass V_DJ_J_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000564" );
    
    public static final OntClass V_VDJ_J_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000565" );
    
    public static final OntClass V_VJ_J_C_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000566" );
    
    public static final OntClass inversion_derived_aneuploid_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000567" );
    
    public static final OntClass bidirectional_promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000568" );
    
    /** <p>GO:0003964 RNA-directed DNA polymerase activity.</p> */
    public static final OntClass retrotransposed = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000569" );
    
    public static final OntClass three_prime_D_recombination_signal_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000570" );
    
    public static final OntClass miRNA_encoding = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000571" );
    
    public static final OntClass DJ_gene_segment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000572" );
    
    public static final OntClass rRNA_encoding = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000573" );
    
    public static final OntClass VDJ_gene_segment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000574" );
    
    public static final OntClass scRNA_encoding = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000575" );
    
    public static final OntClass VJ_gene_segment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000576" );
    
    public static final OntClass centromere = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000577" );
    
    public static final OntClass snoRNA_encoding = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000578" );
    
    public static final OntClass edited_transcript_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000579" );
    
    public static final OntClass methylation_guide_snoRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000580" );
    
    public static final OntClass cap = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000581" );
    
    public static final OntClass rRNA_cleavage_snoRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000582" );
    
    public static final OntClass pre_edited_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000583" );
    
    public static final OntClass tmRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000584" );
    
    public static final OntClass C_D_box_snoRNA_encoding = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000585" );
    
    public static final OntClass tmRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000586" );
    
    /** <p>GO:0000372.</p> */
    public static final OntClass group_I_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000587" );
    
    public static final OntClass autocatalytically_spliced_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000588" );
    
    public static final OntClass SRP_RNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000589" );
    
    public static final OntClass SRP_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000590" );
    
    public static final OntClass pseudoknot = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000591" );
    
    public static final OntClass H_pseudoknot = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000592" );
    
    public static final OntClass C_D_box_snoRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000593" );
    
    public static final OntClass H_ACA_box_snoRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000594" );
    
    public static final OntClass C_D_box_snoRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000595" );
    
    public static final OntClass H_ACA_box_snoRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000596" );
    
    public static final OntClass transcript_edited_by_U_insertion_deletion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000597" );
    
    public static final OntClass edited_by_C_insertion_and_dinucleotide_insertion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000598" );
    
    public static final OntClass edited_by_C_to_U_substitution = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000599" );
    
    public static final OntClass edited_by_A_to_I_substitution = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000600" );
    
    public static final OntClass edited_by_G_addition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000601" );
    
    public static final OntClass guide_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000602" );
    
    /** <p>GO:0000373.</p> */
    public static final OntClass group_II_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000603" );
    
    public static final OntClass editing_block = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000604" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass intergenic_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000605" );
    
    public static final OntClass editing_domain = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000606" );
    
    public static final OntClass unedited_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000607" );
    
    public static final OntClass H_ACA_box_snoRNA_encoding = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000608" );
    
    public static final OntClass oligo_U_tail = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000609" );
    
    public static final OntClass polyA_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000610" );
    
    public static final OntClass branch_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000611" );
    
    public static final OntClass polypyrimidine_tract = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000612" );
    
    public static final OntClass bacterial_RNApol_promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000613" );
    
    public static final OntClass bacterial_terminator = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000614" );
    
    public static final OntClass terminator_of_type_2_RNApol_III_promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000615" );
    
    public static final OntClass transcription_end_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000616" );
    
    public static final OntClass RNApol_III_promoter_type_1 = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000617" );
    
    public static final OntClass RNApol_III_promoter_type_2 = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000618" );
    
    /** <p>Binds TFIIIC.</p> */
    public static final OntClass A_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000619" );
    
    /** <p>Binds TFIIIC.</p> */
    public static final OntClass B_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000620" );
    
    public static final OntClass RNApol_III_promoter_type_3 = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000621" );
    
    public static final OntClass C_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000622" );
    
    public static final OntClass snRNA_encoding = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000623" );
    
    public static final OntClass telomere = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000624" );
    
    public static final OntClass silencer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000625" );
    
    public static final OntClass chromosomal_regulatory_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000626" );
    
    public static final OntClass insulator = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000627" );
    
    public static final OntClass chromosomal_structural_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000628" );
    
    public static final OntClass five_prime_open_reading_frame = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000629" );
    
    public static final OntClass upstream_AUG_codon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000630" );
    
    public static final OntClass polycistronic_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000631" );
    
    public static final OntClass monocistronic_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000632" );
    
    public static final OntClass monocistronic_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000633" );
    
    public static final OntClass polycistronic_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000634" );
    
    public static final OntClass mini_exon_donor_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000635" );
    
    public static final OntClass spliced_leader_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000636" );
    
    public static final OntClass engineered_plasmid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000637" );
    
    public static final OntClass transcribed_spacer_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000638" );
    
    public static final OntClass internal_transcribed_spacer_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000639" );
    
    public static final OntClass external_transcribed_spacer_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000640" );
    
    public static final OntClass tetranucleotide_repeat_microsatellite_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000641" );
    
    public static final OntClass SRP_RNA_encoding = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000642" );
    
    public static final OntClass minisatellite = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000643" );
    
    public static final OntClass antisense_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000644" );
    
    public static final OntClass antisense_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000645" );
    
    public static final OntClass siRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000646" );
    
    public static final OntClass miRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000647" );
    
    public static final OntClass small_subunit_rRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000650" );
    
    public static final OntClass large_subunit_rRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000651" );
    
    public static final OntClass rRNA_5S = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000652" );
    
    public static final OntClass rRNA_28S = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000653" );
    
    public static final OntClass maxicircle_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000654" );
    
    /** <p>A ncRNA is a processed_transcript, so it may not contain parts such as transcribed_spacer_regions 
     *  that are removed in the act of processing. For the corresponding primary_transcripts, 
     *  please see term SO:0000483 nc_primary_transcript.</p>
     */
    public static final OntClass ncRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000655" );
    
    public static final OntClass stRNA_encoding = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000656" );
    
    public static final OntClass repeat_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000657" );
    
    public static final OntClass dispersed_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000658" );
    
    public static final OntClass tmRNA_encoding = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000659" );
    
    public static final OntClass DNA_invertase_target_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000660" );
    
    public static final OntClass intron_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000661" );
    
    /** <p>GO:0000398.</p> */
    public static final OntClass spliceosomal_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000662" );
    
    public static final OntClass tRNA_encoding = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000663" );
    
    public static final OntClass introgressed_chromosome_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000664" );
    
    public static final OntClass monocistronic_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000665" );
    
    public static final OntClass mobile_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000666" );
    
    public static final OntClass insertion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000667" );
    
    public static final OntClass EST_match = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000668" );
    
    public static final OntClass sequence_rearrangement_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000669" );
    
    public static final OntClass chromosome_breakage_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000670" );
    
    public static final OntClass internal_eliminated_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000671" );
    
    public static final OntClass macronucleus_destined_segment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000672" );
    
    public static final OntClass transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000673" );
    
    public static final OntClass non_canonical_splice_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000674" );
    
    public static final OntClass canonical_splice_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000675" );
    
    public static final OntClass canonical_three_prime_splice_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000676" );
    
    public static final OntClass canonical_five_prime_splice_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000677" );
    
    public static final OntClass non_canonical_three_prime_splice_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000678" );
    
    public static final OntClass non_canonical_five_prime_splice_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000679" );
    
    public static final OntClass non_canonical_start_codon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000680" );
    
    public static final OntClass aberrant_processed_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000681" );
    
    public static final OntClass splicing_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000682" );
    
    public static final OntClass exonic_splice_enhancer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000683" );
    
    public static final OntClass nuclease_sensitive_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000684" );
    
    public static final OntClass DNAseI_hypersensitive_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000685" );
    
    public static final OntClass translocation_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000686" );
    
    public static final OntClass deletion_junction = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000687" );
    
    public static final OntClass golden_path = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000688" );
    
    public static final OntClass cDNA_match = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000689" );
    
    public static final OntClass gene_with_polycistronic_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000690" );
    
    public static final OntClass cleaved_initiator_methionine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000691" );
    
    public static final OntClass gene_with_dicistronic_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000692" );
    
    public static final OntClass gene_with_recoded_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000693" );
    
    public static final OntClass SNP = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000694" );
    
    /** <p>Requested by Lynn Crosby, jan 2006.</p> */
    public static final OntClass reagent = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000695" );
    
    public static final OntClass oligo = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000696" );
    
    public static final OntClass gene_with_stop_codon_read_through = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000697" );
    
    public static final OntClass gene_with_stop_codon_redefined_as_pyrrolysine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000698" );
    
    /** <p>A junction is a boundary between regions. A boundary has an extent of zero.</p> */
    public static final OntClass junction = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000699" );
    
    public static final OntClass remark_CLASS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000700" );
    
    public static final OntClass possible_base_call_error = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000701" );
    
    public static final OntClass possible_assembly_error = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000702" );
    
    public static final OntClass experimental_result_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000703" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology. 
     *  A gene may be considered as a unit of inheritance.</p>
     */
    public static final OntClass gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000704" );
    
    public static final OntClass tandem_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000705" );
    
    /** <p>This region contains a polypyridine tract and AG dinucleotide in some organisms 
     *  and is UUUCAG in C. elegans.</p>
     */
    public static final OntClass trans_splice_acceptor_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000706" );
    
    /** <p>SL RNA contains a donor site.</p> */
    public static final OntClass trans_splice_donor_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000707" );
    
    public static final OntClass SL1_acceptor_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000708" );
    
    public static final OntClass SL2_acceptor_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000709" );
    
    public static final OntClass gene_with_stop_codon_redefined_as_selenocysteine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000710" );
    
    public static final OntClass gene_with_mRNA_recoded_by_translational_bypass = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000711" );
    
    public static final OntClass gene_with_transcript_with_translational_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000712" );
    
    public static final OntClass DNA_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000713" );
    
    public static final OntClass nucleotide_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000714" );
    
    public static final OntClass RNA_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000715" );
    
    public static final OntClass dicistronic_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000716" );
    
    /** <p>This term was added after a request by SGD. August 2004. Modified after SO 
     *  meeting in Cambridge to not include start or stop.</p>
     */
    public static final OntClass reading_frame = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000717" );
    
    /** <p>Term requested by Rama from SGD.</p> */
    public static final OntClass blocked_reading_frame = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000718" );
    
    public static final OntClass ultracontig = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000719" );
    
    /** <p>requested by Michael on 19 Nov 2004.</p> */
    public static final OntClass foreign_transposable_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000720" );
    
    /** <p>Requested by Michael, 19 nov 2004.</p> */
    public static final OntClass gene_with_dicistronic_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000721" );
    
    /** <p>Requested by MA nov 19 2004.</p> */
    public static final OntClass gene_with_dicistronic_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000722" );
    
    public static final OntClass iDNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000723" );
    
    public static final OntClass oriT = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000724" );
    
    /** <p>Added to bring SO inline with the EMBL, DDBJ, GenBank feature table. Old definition 
     *  before biosapiens: The coding sequence for an N-terminal domain of a nuclear-encoded 
     *  organellar protein. This domain is involved in post translational import of 
     *  the protein into the organelle.</p>
     */
    public static final OntClass transit_peptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000725" );
    
    /** <p>Added to comply with the feature table. A single repeat.</p> */
    public static final OntClass repeat_unit = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000726" );
    
    /** <p>Requested by Stephen Grossmann Dec 2004.</p> */
    public static final OntClass CRM = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000727" );
    
    /** <p>Intein-mediated protein splicing occurs after mRNA has been translated into 
     *  a protein.</p>
     */
    public static final OntClass intein = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000728" );
    
    public static final OntClass intein_containing = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000729" );
    
    public static final OntClass gap = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000730" );
    
    /** <p>Term added because of request by MO people.</p> */
    public static final OntClass fragmentary = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000731" );
    
    public static final OntClass predicted = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000732" );
    
    public static final OntClass feature_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000733" );
    
    /** <p>Added for the MO people.</p> */
    public static final OntClass exemplar_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000734" );
    
    public static final OntClass sequence_location = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000735" );
    
    public static final OntClass organelle_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000736" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass mitochondrial_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000737" );
    
    public static final OntClass nuclear_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000738" );
    
    public static final OntClass nucleomorphic_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000739" );
    
    public static final OntClass plastid_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000740" );
    
    public static final OntClass kinetoplast = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000741" );
    
    public static final OntClass maxicircle = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000742" );
    
    public static final OntClass apicoplast_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000743" );
    
    public static final OntClass chromoplast_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000744" );
    
    public static final OntClass chloroplast_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000745" );
    
    public static final OntClass cyanelle_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000746" );
    
    public static final OntClass leucoplast_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000747" );
    
    public static final OntClass proplastid_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000748" );
    
    public static final OntClass plasmid_location = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000749" );
    
    public static final OntClass amplification_origin = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000750" );
    
    public static final OntClass proviral_location = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000751" );
    
    public static final OntClass gene_group_regulatory_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000752" );
    
    public static final OntClass clone_insert = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000753" );
    
    public static final OntClass lambda_vector = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000754" );
    
    public static final OntClass plasmid_vector = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000755" );
    
    public static final OntClass cDNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000756" );
    
    public static final OntClass single_stranded_cDNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000757" );
    
    public static final OntClass double_stranded_cDNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000758" );
    
    public static final OntClass plasmid_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000759" );
    
    public static final OntClass YAC_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000760" );
    
    public static final OntClass phagemid_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000761" );
    
    public static final OntClass PAC_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000762" );
    
    public static final OntClass fosmid_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000763" );
    
    public static final OntClass BAC_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000764" );
    
    public static final OntClass cosmid_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000765" );
    
    public static final OntClass pyrrolysyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000766" );
    
    public static final OntClass clone_insert_start_CLASS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000767" );
    
    public static final OntClass episome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000768" );
    
    /** <p>Added in response to comment from Kelly Williams from Indiana. Nov 2005.</p> */
    public static final OntClass tmRNA_coding_piece = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000769" );
    
    /** <p>Added in response to Kelly Williams from Indiana. Date: Nov 2005.</p> */
    public static final OntClass tmRNA_acceptor_piece = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000770" );
    
    /** <p>Added in respose to request by Simon Twigger November 14th 2005.</p> */
    public static final OntClass QTL = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000771" );
    
    /** <p>Genomic islands are transmissible elements characterized by large size (&gt;10kb).</p> */
    public static final OntClass genomic_island = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000772" );
    
    /** <p>Nature Reviews Microbiology 2, 414-424 (2004); doi:10.1038 micro 884 GENOMIC 
     *  ISLANDS IN PATHOGENIC AND ENVIRONMENTAL MICROORGANISMS Ulrich Dobrindt, Bianca 
     *  Hochhut, Ute Hentschel &amp; Jorg Hacker.</p>
     */
    public static final OntClass pathogenic_island = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000773" );
    
    /** <p>Genes for phenolic compound degradation in Pseudomonas putida are found on 
     *  metabolic islands.</p>
     */
    public static final OntClass metabolic_island = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000774" );
    
    /** <p>The iron-uptake ability of many pathogens are conveyed by adaptive islands. 
     *  Nature Reviews Microbiology 2, 414-424 (2004); doi:10.1038 micro 884 GENOMIC 
     *  ISLANDS IN PATHOGENIC AND ENVIRONMENTAL MICROORGANISMS Ulrich Dobrindt, Bianca 
     *  Hochhut, Ute Hentschel &amp; Jorg Hacker.</p>
     */
    public static final OntClass adaptive_island = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000775" );
    
    /** <p>Nitrogen fixation in Rhizobiaceae species is encoded by symbiosis islands. 
     *  Evolution of rhizobia by acquisition of a 500-kb symbiosis island that integrates 
     *  into a phe-tRNA gene. John T. Sullivan and Clive W. Ronso PNAS 1998 Apr 28 
     *  95 (9) 5145-5149.</p>
     */
    public static final OntClass symbiosis_island = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000776" );
    
    /** <p>Added Jan 2006 to allow the annotation of the pseudogenic rRNA by flybase. 
     *  Non-functional is defined as its transcription is prevented due to one or 
     *  more mutatations.</p>
     */
    public static final OntClass pseudogenic_rRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000777" );
    
    /** <p>Added Jan 2006 to allow the annotation of the pseudogenic tRNA by flybase. 
     *  Non-functional is defined as its transcription is prevented due to one or 
     *  more mutatations.</p>
     */
    public static final OntClass pseudogenic_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000778" );
    
    /** <p>Requested by Lynn Crosby Jan 2006.</p> */
    public static final OntClass engineered_episome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000779" );
    
    /** <p>Added by KE Jan 2006 to capture the kinds of attributes of TEs</p> */
    public static final OntClass transposable_element_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000780" );
    
    public static final OntClass transgenic = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000781" );
    
    public static final OntClass natural = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000782" );
    
    public static final OntClass engineered = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000783" );
    
    public static final OntClass foreign = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000784" );
    
    /** <p>Added in response to Lynn Crosby. A clone insert may be composed of many cloned 
     *  regions.</p>
     */
    public static final OntClass cloned_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000785" );
    
    /** <p>Added jan 2006 by KE.</p> */
    public static final OntClass reagent_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000786" );
    
    public static final OntClass clone_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000787" );
    
    public static final OntClass cloned = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000788" );
    
    public static final OntClass validated = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000789" );
    
    public static final OntClass invalidated = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000790" );
    
    public static final OntClass cloned_genomic = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000791" );
    
    public static final OntClass cloned_cDNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000792" );
    
    public static final OntClass engineered_DNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000793" );
    
    public static final OntClass engineered_rescue_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000794" );
    
    public static final OntClass rescue_mini_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000795" );
    
    /** <p>Modified as requested by Lynn - FB. May 2007.</p> */
    public static final OntClass transgenic_transposable_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000796" );
    
    public static final OntClass natural_transposable_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000797" );
    
    public static final OntClass engineered_transposable_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000798" );
    
    public static final OntClass engineered_foreign_transposable_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000799" );
    
    public static final OntClass assortment_derived_duplication_CLASS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000800" );
    
    public static final OntClass assortment_derived_deficiency_plus_duplication_CLASS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000801" );
    
    public static final OntClass assortment_derived_deficiency_CLASS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000802" );
    
    public static final OntClass assortment_derived_aneuploid_CLASS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000803" );
    
    public static final OntClass engineered_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000804" );
    
    public static final OntClass engineered_foreign_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000805" );
    
    public static final OntClass fusion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000806" );
    
    public static final OntClass engineered_tag = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000807" );
    
    public static final OntClass validated_cDNA_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000808" );
    
    public static final OntClass invalidated_cDNA_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000809" );
    
    public static final OntClass chimeric_cDNA_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000810" );
    
    public static final OntClass genomically_contaminated_cDNA_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000811" );
    
    public static final OntClass polyA_primed_cDNA_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000812" );
    
    public static final OntClass partially_processed_cDNA_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000813" );
    
    public static final OntClass rescue = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000814" );
    
    public static final OntClass mini_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000815" );
    
    public static final OntClass rescue_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000816" );
    
    public static final OntClass wild_type = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000817" );
    
    public static final OntClass wild_type_rescue_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000818" );
    
    public static final OntClass mitochondrial_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000819" );
    
    public static final OntClass chloroplast_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000820" );
    
    public static final OntClass chromoplast_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000821" );
    
    public static final OntClass cyanelle_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000822" );
    
    public static final OntClass leucoplast_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000823" );
    
    public static final OntClass macronuclear_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000824" );
    
    public static final OntClass micronuclear_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000825" );
    
    public static final OntClass nuclear_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000828" );
    
    public static final OntClass nucleomorphic_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000829" );
    
    /** <p>This is a manufactured term, that serves the purpose of allow the parts of 
     *  a chromosome to have an is_a path to the root.</p>
     */
    public static final OntClass chromosome_part = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000830" );
    
    /** <p>A manufactured term used to allow the parts of a gene to have an is_a path 
     *  to the root.</p>
     */
    public static final OntClass gene_member_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000831" );
    
    /** <p>This is a manufactured term to allow the parts of promoter to have an is_a 
     *  path back to the root.</p>
     */
    public static final OntClass promoter_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000832" );
    
    /** <p>This term was added to provide a grouping term for the region parts of transcript, 
     *  thus giving them an is_a path back to the root.</p>
     */
    public static final OntClass transcript_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000833" );
    
    /** <p>A manufactured term to collect together the parts of a mature transcript and 
     *  give them an is_a path to the root.</p>
     */
    public static final OntClass mature_transcript_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000834" );
    
    /** <p>This term was added to provide a grouping term for the region parts of primary_transcript, 
     *  thus giving them an is_a path back to the root.</p>
     */
    public static final OntClass primary_transcript_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000835" );
    
    /** <p>This term was added to provide a grouping term for the region parts of mRNA, 
     *  thus giving them an is_a path back to the root.</p>
     */
    public static final OntClass mRNA_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000836" );
    
    /** <p>A region of UTR. This term is a grouping term to allow the parts of UTR to 
     *  have an is_a path to the root.</p>
     */
    public static final OntClass UTR_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000837" );
    
    /** <p>To allow transcribed_spacer_region to have a path to the root.</p> */
    public static final OntClass rRNA_primary_transcript_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000838" );
    
    /** <p>Added to allow the polypeptide regions to have is_a paths back to the root.</p> */
    public static final OntClass polypeptide_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000839" );
    
    /** <p>A manufactured to group the parts of repeats, to give them an is_a path back 
     *  to the root.</p>
     */
    public static final OntClass repeat_component = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000840" );
    
    /** <p>A terms added to allow the parts of introns to have is_a paths to the root.</p> */
    public static final OntClass spliceosomal_intron_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000841" );
    
    public static final OntClass gene_component_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000842" );
    
    /** <p>This is a manufactured term to allow the parts of bacterial_RNApol_promoter 
     *  to have an is_a path back to the root.</p>
     */
    public static final OntClass bacterial_RNApol_promoter_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000843" );
    
    /** <p>This is a manufactured term to allow the parts of RNApol_II_promoter to have 
     *  an is_a path back to the root.</p>
     */
    public static final OntClass RNApol_II_promoter_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000844" );
    
    /** <p>This is a manufactured term to allow the parts of RNApol_III_promoter_type_1 
     *  to have an is_a path back to the root.</p>
     */
    public static final OntClass RNApol_III_promoter_type_1_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000845" );
    
    /** <p>This is a manufactured term to allow the parts of RNApol_III_promoter_type_2 
     *  to have an is_a path back to the root.</p>
     */
    public static final OntClass RNApol_III_promoter_type_2_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000846" );
    
    /** <p>This term was added to provide a grouping term for the region parts of tmRNA, 
     *  thus giving them an is_a path back to the root.</p>
     */
    public static final OntClass tmRNA_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000847" );
    
    public static final OntClass LTR_component = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000848" );
    
    public static final OntClass three_prime_LTR_component = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000849" );
    
    public static final OntClass five_prime_LTR_component = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000850" );
    
    public static final OntClass CDS_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000851" );
    
    public static final OntClass exon_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000852" );
    
    public static final OntClass homologous_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000853" );
    
    /** <p>A term to be used in conjunction with the paralogous_to relationship.</p> */
    public static final OntClass paralogous_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000854" );
    
    /** <p>This term should be used in conjunction with the similarity relationships 
     *  defined in SO.</p>
     */
    public static final OntClass orthologous_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000855" );
    
    public static final OntClass conserved = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000856" );
    
    public static final OntClass homologous = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000857" );
    
    public static final OntClass orthologous = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000858" );
    
    public static final OntClass paralogous = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000859" );
    
    public static final OntClass syntenic = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000860" );
    
    public static final OntClass capped_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000861" );
    
    public static final OntClass capped_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000862" );
    
    public static final OntClass mRNA_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000863" );
    
    public static final OntClass exemplar = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000864" );
    
    public static final OntClass frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000865" );
    
    public static final OntClass minus_1_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000866" );
    
    public static final OntClass minus_2_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000867" );
    
    public static final OntClass plus_1_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000868" );
    
    public static final OntClass plus_2_framshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000869" );
    
    public static final OntClass trans_spliced = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000870" );
    
    public static final OntClass polyadenylated_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000871" );
    
    public static final OntClass trans_spliced_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000872" );
    
    public static final OntClass edited_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000873" );
    
    public static final OntClass edited_transcript_by_A_to_I_substitution = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000874" );
    
    public static final OntClass bound_by_protein = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000875" );
    
    public static final OntClass bound_by_nucleic_acid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000876" );
    
    public static final OntClass alternatively_spliced = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000877" );
    
    public static final OntClass monocistronic = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000878" );
    
    public static final OntClass dicistronic = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000879" );
    
    public static final OntClass polycistronic = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000880" );
    
    public static final OntClass recoded = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000881" );
    
    public static final OntClass codon_redefined = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000882" );
    
    public static final OntClass stop_codon_read_through = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000883" );
    
    public static final OntClass stop_codon_redefined_as_pyrrolysine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000884" );
    
    public static final OntClass stop_codon_redefined_as_selenocysteine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000885" );
    
    public static final OntClass recoded_by_translational_bypass = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000886" );
    
    public static final OntClass translationally_frameshifted = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000887" );
    
    public static final OntClass maternally_imprinted_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000888" );
    
    public static final OntClass paternally_imprinted_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000889" );
    
    public static final OntClass post_translationally_regulated_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000890" );
    
    public static final OntClass negatively_autoregulated_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000891" );
    
    public static final OntClass positively_autoregulated_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000892" );
    
    public static final OntClass silenced = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000893" );
    
    public static final OntClass silenced_by_DNA_modification = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000894" );
    
    public static final OntClass silenced_by_DNA_methylation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000895" );
    
    public static final OntClass translationally_regulated_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000896" );
    
    public static final OntClass allelically_excluded_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000897" );
    
    public static final OntClass epigenetically_modified_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000898" );
    
    public static final OntClass nuclear_mitochondrial = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000899" );
    
    public static final OntClass processed = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000900" );
    
    public static final OntClass unequally_crossed_over = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000901" );
    
    public static final OntClass transgene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000902" );
    
    public static final OntClass endogenous_retroviral_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000903" );
    
    public static final OntClass rearranged_at_DNA_level = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000904" );
    
    /** <p>This term is the hypernym of attributes and should not be annotated to.</p> */
    public static final OntClass status = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000905" );
    
    public static final OntClass independently_known = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000906" );
    
    public static final OntClass supported_by_sequence_similarity = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000907" );
    
    public static final OntClass supported_by_domain_match = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000908" );
    
    public static final OntClass supported_by_EST_or_cDNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000909" );
    
    public static final OntClass orphan = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000910" );
    
    public static final OntClass predicted_by_ab_initio_computation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000911" );
    
    public static final OntClass asx_turn = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000912" );
    
    public static final OntClass cloned_cDNA_insert = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000913" );
    
    public static final OntClass cloned_genomic_insert = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000914" );
    
    public static final OntClass engineered_insert = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000915" );
    
    public static final OntClass edit_operation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000916" );
    
    /** <p>The insertion and deletion of uridine (U) residues, usually within coding 
     *  regions of mRNA transcripts of cryptogenes in the mitochondrial genome of 
     *  kinetoplastid protozoa.</p>
     */
    public static final OntClass insert_U = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000917" );
    
    /** <p>The insertion and deletion of uridine (U) residues, usually within coding 
     *  regions of mRNA transcripts of cryptogenes in the mitochondrial genome of 
     *  kinetoplastid protozoa.</p>
     */
    public static final OntClass delete_U = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000918" );
    
    public static final OntClass substitute_A_to_I = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000919" );
    
    public static final OntClass insert_C = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000920" );
    
    public static final OntClass insert_dinucleotide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000921" );
    
    public static final OntClass substitute_C_to_U = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000922" );
    
    public static final OntClass insert_G = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000923" );
    
    /** <p>The type of RNA editing found in the mitochondria of Myxomycota, characterized 
     *  by the insertion of mono- and dinucleotides in RNAs relative to their mtDNA 
     *  template and in addition, C to U base conversion. The most common mononucleotide 
     *  insertion is cytidine, although a number of uridine mononucleotides are inserted 
     *  at specific sites. Adenine and guanine have not been observed in mononucleotide 
     *  insertions. Five different dinucleotide insertions have been observed, GC, 
     *  GU, CU, AU and AA. Both mono- and dinucleotide insertions create open reading 
     *  frames in mRNA and contribute to highly conserved structural features of rRNAs 
     *  and tRNAs.</p>
     */
    public static final OntClass insert_GC = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000924" );
    
    /** <p>The type of RNA editing found in the mitochondria of Myxomycota, characterized 
     *  by the insertion of mono- and dinucleotides in RNAs relative to their mtDNA 
     *  template and in addition, C to U base conversion. The most common mononucleotide 
     *  insertion is cytidine, although a number of uridine mononucleotides are inserted 
     *  at specific sites. Adenine and guanine have not been observed in mononucleotide 
     *  insertions. Five different dinucleotide insertions have been observed, GC, 
     *  GU, CU, AU and AA. Both mono- and dinucleotide insertions create open reading 
     *  frames in mRNA and contribute to highly conserved structural features of rRNAs 
     *  and tRNAs.</p>
     */
    public static final OntClass insert_GU = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000925" );
    
    /** <p>The type of RNA editing found in the mitochondria of Myxomycota, characterized 
     *  by the insertion of mono- and dinucleotides in RNAs relative to their mtDNA 
     *  template and in addition, C to U base conversion. The most common mononucleotide 
     *  insertion is cytidine, although a number of uridine mononucleotides are inserted 
     *  at specific sites. Adenine and guanine have not been observed in mononucleotide 
     *  insertions. Five different dinucleotide insertions have been observed, GC, 
     *  GU, CU, AU and AA. Both mono- and dinucleotide insertions create open reading 
     *  frames in mRNA and contribute to highly conserved structural features of rRNAs 
     *  and tRNAs.</p>
     */
    public static final OntClass insert_CU = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000926" );
    
    /** <p>The type of RNA editing found in the mitochondria of Myxomycota, characterized 
     *  by the insertion of mono- and dinucleotides in RNAs relative to their mtDNA 
     *  template and in addition, C to U base conversion. The most common mononucleotide 
     *  insertion is cytidine, although a number of uridine mononucleotides are inserted 
     *  at specific sites. Adenine and guanine have not been observed in mononucleotide 
     *  insertions. Five different dinucleotide insertions have been observed, GC, 
     *  GU, CU, AU and AA. Both mono- and dinucleotide insertions create open reading 
     *  frames in mRNA and contribute to highly conserved structural features of rRNAs 
     *  and tRNAs.</p>
     */
    public static final OntClass insert_AU = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000927" );
    
    /** <p>The type of RNA editing found in the mitochondria of Myxomycota, characterized 
     *  by the insertion of mono- and dinucleotides in RNAs relative to their mtDNA 
     *  template and in addition, C to U base conversion. The most common mononucleotide 
     *  insertion is cytidine, although a number of uridine mononucleotides are inserted 
     *  at specific sites. Adenine and guanine have not been observed in mononucleotide 
     *  insertions. Five different dinucleotide insertions have been observed, GC, 
     *  GU, CU, AU and AA. Both mono- and dinucleotide insertions create open reading 
     *  frames in mRNA and contribute to highly conserved structural features of rRNAs 
     *  and tRNAs.</p>
     */
    public static final OntClass insert_AA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000928" );
    
    public static final OntClass edited_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000929" );
    
    public static final OntClass guide_RNA_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000930" );
    
    public static final OntClass anchor_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000931" );
    
    public static final OntClass pre_edited_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000932" );
    
    public static final OntClass intermediate = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000933" );
    
    public static final OntClass miRNA_target_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000934" );
    
    public static final OntClass edited_CDS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000935" );
    
    public static final OntClass vertebrate_immunoglobulin_T_cell_receptor_rearranged_segment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000936" );
    
    public static final OntClass vertebrate_immune_system_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000937" );
    
    public static final OntClass vertebrate_immunoglobulin_T_cell_receptor_rearranged_gene_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000938" );
    
    public static final OntClass vertebrate_immune_system_gene_recombination_signal_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000939" );
    
    public static final OntClass recombinationally_rearranged = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000940" );
    
    public static final OntClass recombinationally_rearranged_vertebrate_immune_system_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000941" );
    
    public static final OntClass attP_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000942" );
    
    public static final OntClass attB_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000943" );
    
    public static final OntClass attL_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000944" );
    
    public static final OntClass attR_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000945" );
    
    public static final OntClass integration_excision_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000946" );
    
    public static final OntClass resolution_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000947" );
    
    /** <p>A target region for site-specific inversion of a DNA region and which carries 
     *  binding sites for a site-specific recombinase and accessory proteins as well 
     *  as the site for specific cleavage by the recombinase.</p>
     */
    public static final OntClass inversion_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000948" );
    
    public static final OntClass dif_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000949" );
    
    public static final OntClass attC_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000950" );
    
    public static final OntClass eukaryotic_terminator = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000951" );
    
    public static final OntClass oriV = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000952" );
    
    public static final OntClass oriC = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000953" );
    
    public static final OntClass DNA_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000954" );
    
    public static final OntClass double_stranded_DNA_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000955" );
    
    public static final OntClass single_stranded_DNA_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000956" );
    
    public static final OntClass linear_double_stranded_DNA_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000957" );
    
    public static final OntClass circular_double_stranded_DNA_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000958" );
    
    public static final OntClass linear_single_stranded_DNA_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000959" );
    
    public static final OntClass circular_single_stranded_DNA_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000960" );
    
    public static final OntClass RNA_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000961" );
    
    public static final OntClass single_stranded_RNA_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000962" );
    
    public static final OntClass linear_single_stranded_RNA_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000963" );
    
    public static final OntClass linear_double_stranded_RNA_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000964" );
    
    public static final OntClass double_stranded_RNA_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000965" );
    
    public static final OntClass circular_single_stranded_RNA_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000966" );
    
    public static final OntClass circular_double_stranded_RNA_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000967" );
    
    /** <p>This has been obsoleted as it represents a process. replaced_by: GO:0034961.</p> */
    public static final OntClass sequence_replication_mode = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000968" );
    
    /** <p>This has been obsoleted as it represents a process. replaced_by: GO:0070581.</p> */
    public static final OntClass rolling_circle = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000969" );
    
    /** <p>This has been obsoleted as it represents a process. replaced_by: GO:0070582</p> */
    public static final OntClass theta_replication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000970" );
    
    /** <p>This has been obsoleted as it represents a process. replaced_by: GO:0006260.</p> */
    public static final OntClass DNA_replication_mode = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000971" );
    
    /** <p>This has been obsoleted as it represents a process. replaced_by: GO:0034961.</p> */
    public static final OntClass RNA_replication_mode = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000972" );
    
    public static final OntClass insertion_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000973" );
    
    public static final OntClass minicircle_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000975" );
    
    public static final OntClass cryptic = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000976" );
    
    /** <p>Part of an edited transcript only.</p> */
    public static final OntClass anchor_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000977" );
    
    public static final OntClass template_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000978" );
    
    public static final OntClass gRNA_encoding = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000979" );
    
    public static final OntClass minicircle = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000980" );
    
    public static final OntClass rho_dependent_bacterial_terminator = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000981" );
    
    public static final OntClass rho_independent_bacterial_terminator = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000982" );
    
    /** <p>Attributes added to describe the different kinds of replicon. SO workshop, 
     *  September 2006.</p>
     */
    public static final OntClass strand_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000983" );
    
    /** <p>Attributes added to describe the different kinds of replicon. SO workshop, 
     *  September 2006.</p>
     */
    public static final OntClass single = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000984" );
    
    /** <p>Attributes added to describe the different kinds of replicon. SO workshop, 
     *  September 2006.</p>
     */
    public static final OntClass double_ = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000985" );
    
    /** <p>Attributes added to describe the different kinds of replicon. SO workshop, 
     *  September 2006.</p>
     */
    public static final OntClass topology_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000986" );
    
    /** <p>Attributes added to describe the different kinds of replicon. SO workshop, 
     *  September 2006.</p>
     */
    public static final OntClass linear = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000987" );
    
    /** <p>Attributes added to describe the different kinds of replicon. SO workshop, 
     *  September 2006.</p>
     */
    public static final OntClass circular = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000988" );
    
    public static final OntClass class_II_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000989" );
    
    /** <p>Requested by Karen Pilcher - Dictybase. song-Term Tracker-1574577.</p> */
    public static final OntClass class_I_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000990" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass genomic_DNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000991" );
    
    /** <p>Requested by Andy Schroder - Flybase Harvard, Nov 2006.</p> */
    public static final OntClass BAC_cloned_genomic_insert = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000992" );
    
    /** <p>Term added Dec 06 to comply with mapping to MGED terms. It should be used 
     *  to generate consensus regions. The specific cross product terms they require 
     *  are consensus_region and consensus_mRNA.</p>
     */
    public static final OntClass consensus = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000993" );
    
    /** <p>DO not obsolete without considering MGED mapping.</p> */
    public static final OntClass consensus_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000994" );
    
    /** <p>DO not obsolete without considering MGED mapping.</p> */
    public static final OntClass consensus_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000995" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass predicted_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000996" );
    
    /** <p>This term is mapped to MGED. Do not obsolete without consulting MGED ontology.</p> */
    public static final OntClass gene_fragment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000997" );
    
    public static final OntClass recursive_splice_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000998" );
    
    /** <p>Requested by Keith Boroevich December, 2006.</p> */
    public static final OntClass BAC_end = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0000999" );
    
    public static final OntClass rRNA_16S = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001000" );
    
    public static final OntClass rRNA_23S = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001001" );
    
    public static final OntClass rRNA_25S = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001002" );
    
    /** <p>Requested by Hadi Quesneville January 2007.</p> */
    public static final OntClass solo_LTR = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001003" );
    
    public static final OntClass low_complexity = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001004" );
    
    public static final OntClass low_complexity_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001005" );
    
    public static final OntClass prophage = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001006" );
    
    /** <p>This is not cryptic in the same sense as a cryptic gene or cryptic splice 
     *  site.</p>
     */
    public static final OntClass cryptic_prophage = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001007" );
    
    public static final OntClass tetraloop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001008" );
    
    public static final OntClass DNA_constraint_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001009" );
    
    public static final OntClass i_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001010" );
    
    public static final OntClass PNA_oligo = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001011" );
    
    /** <p>Added by request from Colin Batchelor.</p> */
    public static final OntClass DNAzyme = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001012" );
    
    public static final OntClass MNP = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001013" );
    
    /** <p>Requested by Colin Batchelor, Feb 2007.</p> */
    public static final OntClass intron_domain = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001014" );
    
    public static final OntClass wobble_base_pair = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001015" );
    
    public static final OntClass internal_guide_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001016" );
    
    /** <p>Added in March 2007 in after meeting with PharmGKB. Although this term is 
     *  in common usage, it is better to annotate with the most specific term possible, 
     *  such as synonymous codon, intron variant etc.</p>
     */
    public static final OntClass silent_mutation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001017" );
    
    /** <p>Requested by Trish Whetzel.</p> */
    public static final OntClass epitope = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001018" );
    
    public static final OntClass copy_number_variation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001019" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_copy_number = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001020" );
    
    public static final OntClass chromosome_breakpoint = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001021" );
    
    public static final OntClass inversion_breakpoint = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001022" );
    
    public static final OntClass allele = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001023" );
    
    public static final OntClass haplotype = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001024" );
    
    public static final OntClass polymorphic_sequence_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001025" );
    
    public static final OntClass genome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001026" );
    
    public static final OntClass genotype = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001027" );
    
    public static final OntClass diplotype = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001028" );
    
    public static final OntClass direction_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001029" );
    
    public static final OntClass forward = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001030" );
    
    public static final OntClass reverse = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001031" );
    
    /** <p>This terms is used by MO.</p> */
    public static final OntClass mitochondrial_DNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001032" );
    
    /** <p>This term is used by MO.</p> */
    public static final OntClass chloroplast_DNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001033" );
    
    /** <p>Ruby et al. Nature 448:83 describe a new class of miRNAs that are derived 
     *  from de-branched introns.</p>
     */
    public static final OntClass miRtron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001034" );
    
    public static final OntClass piRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001035" );
    
    public static final OntClass arginyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001036" );
    
    public static final OntClass mobile_genetic_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001037" );
    
    public static final OntClass extrachromosomal_mobile_genetic_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001038" );
    
    public static final OntClass integrated_mobile_genetic_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001039" );
    
    public static final OntClass integrated_plasmid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001040" );
    
    /** <p>The definitions of the children of this term were revised Decemeber 2007 after 
     *  discussion on song-devel. The resulting definitions are slightly unweildy 
     *  but hopefully more logically correct.</p>
     */
    public static final OntClass viral_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001041" );
    
    public static final OntClass phage_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001042" );
    
    public static final OntClass attCtn_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001043" );
    
    /** <p>Definition change requested by Val, 3172757.</p> */
    public static final OntClass nuclear_mt_pseudogene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001044" );
    
    public static final OntClass cointegrated_plasmid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001045" );
    
    public static final OntClass IRLinv_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001046" );
    
    public static final OntClass IRRinv_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001047" );
    
    /** <p>A term created to allow the parts of an inversion site have an is_a path back 
     *  to the root.</p>
     */
    public static final OntClass inversion_site_part = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001048" );
    
    public static final OntClass defective_conjugative_transposon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001049" );
    
    /** <p>Requested by Chris Smith, and others at Flybase to help annotate nested repeats.</p> */
    public static final OntClass repeat_fragment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001050" );
    
    public static final OntClass nested_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001051" );
    
    public static final OntClass nested_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001052" );
    
    public static final OntClass nested_transposon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001053" );
    
    public static final OntClass transposon_fragment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001054" );
    
    public static final OntClass transcriptional_cis_regulatory_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001055" );
    
    public static final OntClass splicing_regulatory_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001056" );
    
    public static final OntClass enhanceosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001057" );
    
    public static final OntClass promoter_targeting_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001058" );
    
    /** <p>Merged with partially characterized change in nucleotide sequence.</p> */
    public static final OntClass sequence_alteration = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001059" );
    
    public static final OntClass sequence_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001060" );
    
    /** <p>Discrete.</p> */
    public static final OntClass propeptide_cleavage_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001061" );
    
    /** <p>Range.</p> */
    public static final OntClass propeptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001062" );
    
    /** <p>Range.</p> */
    public static final OntClass immature_peptide_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001063" );
    
    /** <p>Hormones, neuropeptides, antimicrobial peptides, are active peptides. They 
     *  are typically short (&lt;40 amino acids) in length.</p>
     */
    public static final OntClass active_peptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001064" );
    
    /** <p>Range.</p> */
    public static final OntClass compositionally_biased_region_of_peptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001066" );
    
    /** <p>Range.</p> */
    public static final OntClass polypeptide_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001067" );
    
    /** <p>Range.</p> */
    public static final OntClass polypeptide_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001068" );
    
    /** <p>Range.</p> */
    public static final OntClass polypeptide_structural_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001070" );
    
    /** <p>Range.</p> */
    public static final OntClass membrane_structure = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001071" );
    
    /** <p>Range.</p> */
    public static final OntClass extramembrane_polypeptide_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001072" );
    
    public static final OntClass cytoplasmic_polypeptide_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001073" );
    
    /** <p>This could be inside an organelle within the cell.</p> */
    public static final OntClass non_cytoplasmic_polypeptide_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001074" );
    
    public static final OntClass intramembrane_polypeptide_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001075" );
    
    public static final OntClass membrane_peptide_loop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001076" );
    
    public static final OntClass transmembrane_polypeptide_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001077" );
    
    /** <p>Biosapien term was secondary_structure.</p> */
    public static final OntClass polypeptide_secondary_structure = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001078" );
    
    public static final OntClass polypeptide_structural_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001079" );
    
    /** <p>Range.</p> */
    public static final OntClass coiled_coil = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001080" );
    
    public static final OntClass helix_turn_helix = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001081" );
    
    /** <p>Range.</p> */
    public static final OntClass polypeptide_sequencing_information = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001082" );
    
    public static final OntClass non_adjacent_residues = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001083" );
    
    /** <p>Discrete.</p> */
    public static final OntClass non_terminal_residue = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001084" );
    
    /** <p>Discrete.</p> */
    public static final OntClass sequence_conflict = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001085" );
    
    public static final OntClass sequence_uncertainty = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001086" );
    
    public static final OntClass cross_link = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001087" );
    
    /** <p>2 discreet &amp; joined.</p> */
    public static final OntClass disulfide_bond = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001088" );
    
    /** <p>Discrete.</p> */
    public static final OntClass post_translationally_modified_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001089" );
    
    public static final OntClass covalent_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001090" );
    
    /** <p>Discrete.</p> */
    public static final OntClass non_covalent_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001091" );
    
    /** <p>Residue is part of a binding site for a metal ion.</p> */
    public static final OntClass polypeptide_metal_contact = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001092" );
    
    public static final OntClass protein_protein_contact = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001093" );
    
    /** <p>Residue involved in contact with calcium.</p> */
    public static final OntClass polypeptide_calcium_ion_contact_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001094" );
    
    public static final OntClass polypeptide_cobalt_ion_contact_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001095" );
    
    public static final OntClass polypeptide_copper_ion_contact_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001096" );
    
    public static final OntClass polypeptide_iron_ion_contact_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001097" );
    
    public static final OntClass polypeptide_magnesium_ion_contact_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001098" );
    
    public static final OntClass polypeptide_manganese_ion_contact_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001099" );
    
    public static final OntClass polypeptide_molybdenum_ion_contact_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001100" );
    
    public static final OntClass polypeptide_nickel_ion_contact_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001101" );
    
    public static final OntClass polypeptide_tungsten_ion_contact_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001102" );
    
    public static final OntClass polypeptide_zinc_ion_contact_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001103" );
    
    /** <p>Discrete.</p> */
    public static final OntClass catalytic_residue = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001104" );
    
    public static final OntClass polypeptide_ligand_contact = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001105" );
    
    public static final OntClass asx_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001106" );
    
    public static final OntClass beta_bulge = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001107" );
    
    public static final OntClass beta_bulge_loop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001108" );
    
    public static final OntClass beta_bulge_loop_five = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001109" );
    
    public static final OntClass beta_bulge_loop_six = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001110" );
    
    /** <p>Range.</p> */
    public static final OntClass beta_strand = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001111" );
    
    /** <p>Range.</p> */
    public static final OntClass antiparallel_beta_strand = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001112" );
    
    /** <p>Range.</p> */
    public static final OntClass parallel_beta_strand = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001113" );
    
    /** <p>Range.</p> */
    public static final OntClass peptide_helix = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001114" );
    
    public static final OntClass left_handed_peptide_helix = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001115" );
    
    public static final OntClass right_handed_peptide_helix = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001116" );
    
    /** <p>Range.</p> */
    public static final OntClass alpha_helix = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001117" );
    
    /** <p>Range.</p> */
    public static final OntClass pi_helix = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001118" );
    
    /** <p>Range.</p> */
    public static final OntClass three_ten_helix = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001119" );
    
    public static final OntClass polypeptide_nest_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001120" );
    
    public static final OntClass polypeptide_nest_left_right_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001121" );
    
    public static final OntClass polypeptide_nest_right_left_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001122" );
    
    public static final OntClass schellmann_loop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001123" );
    
    public static final OntClass schellmann_loop_seven = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001124" );
    
    public static final OntClass schellmann_loop_six = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001125" );
    
    public static final OntClass serine_threonine_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001126" );
    
    public static final OntClass serine_threonine_staple_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001127" );
    
    /** <p>Range.</p> */
    public static final OntClass polypeptide_turn_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001128" );
    
    public static final OntClass asx_turn_left_handed_type_one = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001129" );
    
    public static final OntClass asx_turn_left_handed_type_two = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001130" );
    
    public static final OntClass asx_turn_right_handed_type_two = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001131" );
    
    public static final OntClass asx_turn_right_handed_type_one = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001132" );
    
    public static final OntClass beta_turn = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001133" );
    
    public static final OntClass beta_turn_left_handed_type_one = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001134" );
    
    public static final OntClass beta_turn_left_handed_type_two = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001135" );
    
    public static final OntClass beta_turn_right_handed_type_one = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001136" );
    
    public static final OntClass beta_turn_right_handed_type_two = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001137" );
    
    public static final OntClass gamma_turn = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001138" );
    
    public static final OntClass gamma_turn_classic = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001139" );
    
    public static final OntClass gamma_turn_inverse = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001140" );
    
    public static final OntClass serine_threonine_turn = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001141" );
    
    public static final OntClass st_turn_left_handed_type_one = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001142" );
    
    public static final OntClass st_turn_left_handed_type_two = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001143" );
    
    public static final OntClass st_turn_right_handed_type_one = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001144" );
    
    public static final OntClass st_turn_right_handed_type_two = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001145" );
    
    /** <p>For example, was a substitution natural or mutated as part of an experiment? 
     *  This term is added to merge the biosapiens term sequence_variations.</p>
     */
    public static final OntClass polypeptide_variation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001146" );
    
    /** <p>Discrete.</p> */
    public static final OntClass natural_variant_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001147" );
    
    /** <p>Discrete.</p> */
    public static final OntClass mutated_variant_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001148" );
    
    /** <p>Discrete.</p> */
    public static final OntClass alternate_sequence_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001149" );
    
    public static final OntClass beta_turn_type_six = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001150" );
    
    public static final OntClass beta_turn_type_six_a = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001151" );
    
    public static final OntClass beta_turn_type_six_a_one = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001152" );
    
    public static final OntClass beta_turn_type_six_a_two = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001153" );
    
    public static final OntClass beta_turn_type_six_b = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001154" );
    
    public static final OntClass beta_turn_type_eight = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001155" );
    
    /** <p>This consensus sequence was identified computationally using the MEME algorithm 
     *  within core promoter sequences from -60 to +40, with an E value of 1.7e-183. 
     *  Tends to co-occur with Motif 7. Tends to not occur with DPE motif (SO:0000015) 
     *  or motif 10.</p>
     */
    public static final OntClass DRE_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001156" );
    
    public static final OntClass DMv4_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001157" );
    
    public static final OntClass E_box_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001158" );
    
    public static final OntClass DMv5_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001159" );
    
    public static final OntClass DMv3_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001160" );
    
    public static final OntClass DMv2_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001161" );
    
    public static final OntClass MTE = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001162" );
    
    public static final OntClass INR1_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001163" );
    
    public static final OntClass DPE1_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001164" );
    
    public static final OntClass DMv1_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001165" );
    
    public static final OntClass GAGA_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001166" );
    
    public static final OntClass NDM2_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001167" );
    
    public static final OntClass NDM3_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001168" );
    
    public static final OntClass ds_RNA_viral_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001169" );
    
    public static final OntClass polinton = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001170" );
    
    public static final OntClass rRNA_21S = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001171" );
    
    public static final OntClass tRNA_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001172" );
    
    public static final OntClass anticodon_loop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001173" );
    
    public static final OntClass anticodon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001174" );
    
    public static final OntClass CCA_tail = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001175" );
    
    public static final OntClass DHU_loop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001176" );
    
    public static final OntClass T_loop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001177" );
    
    public static final OntClass pyrrolysine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001178" );
    
    /** <p>The definition is most of the old definition for snoRNA (SO:0000275).</p> */
    public static final OntClass U3_snoRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001179" );
    
    public static final OntClass AU_rich_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001180" );
    
    /** <p>Not to be confused with BRE_motif (SO:0000016), which binds transcription 
     *  factor II B.</p>
     */
    public static final OntClass Bruno_response_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001181" );
    
    public static final OntClass iron_responsive_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001182" );
    
    /** <p>Do not use this for feature annotation. Use morpholino_oligo (SO:0000034) 
     *  instead.</p>
     */
    public static final OntClass morpholino_backbone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001183" );
    
    /** <p>Do not use this term for feature annotation. Use PNA_oligo (SO:0001011) instead.</p> */
    public static final OntClass PNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001184" );
    
    /** <p>Do not use this for feature annotation. Use enzymatic_RNA (SO:0000372) instead.</p> */
    public static final OntClass enzymatic = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001185" );
    
    /** <p>Do not use this for feature annotation. Use ribozyme (SO:0000374) instead.</p> */
    public static final OntClass ribozymic = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001186" );
    
    /** <p>Has RNA pseudouridylation guide activity (GO:0030558).</p> */
    public static final OntClass pseudouridylation_guide_snoRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001187" );
    
    /** <p>Do not use this term for feature annotation. Use LNA_oligo (SO:0001189) instead.</p> */
    public static final OntClass LNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001188" );
    
    public static final OntClass LNA_oligo = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001189" );
    
    /** <p>Do not use this term for feature annotation. Use TNA_oligo (SO:0001191) instead.</p> */
    public static final OntClass TNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001190" );
    
    public static final OntClass TNA_oligo = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001191" );
    
    /** <p>Do not use this term for feature annotation. Use GNA_oligo (SO:0001192) instead.</p> */
    public static final OntClass GNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001192" );
    
    public static final OntClass GNA_oligo = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001193" );
    
    /** <p>Do not use this term for feature annotation. Use R_GNA_oligo (SO:0001195) 
     *  instead.</p>
     */
    public static final OntClass R_GNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001194" );
    
    public static final OntClass R_GNA_oligo = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001195" );
    
    /** <p>Do not use this term for feature annotation. Use S_GNA_oligo (SO:0001197) 
     *  instead.</p>
     */
    public static final OntClass S_GNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001196" );
    
    public static final OntClass S_GNA_oligo = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001197" );
    
    public static final OntClass ds_DNA_viral_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001198" );
    
    public static final OntClass ss_RNA_viral_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001199" );
    
    public static final OntClass negative_sense_ssRNA_viral_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001200" );
    
    public static final OntClass positive_sense_ssRNA_viral_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001201" );
    
    public static final OntClass ambisense_ssRNA_viral_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001202" );
    
    public static final OntClass RNA_polymerase_promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001203" );
    
    public static final OntClass Phage_RNA_Polymerase_Promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001204" );
    
    public static final OntClass SP6_RNA_Polymerase_Promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001205" );
    
    public static final OntClass T3_RNA_Polymerase_Promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001206" );
    
    public static final OntClass T7_RNA_Polymerase_Promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001207" );
    
    public static final OntClass five_prime_EST = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001208" );
    
    public static final OntClass three_prime_EST = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001209" );
    
    public static final OntClass translational_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001210" );
    
    public static final OntClass plus_1_translational_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001211" );
    
    public static final OntClass plus_2_translational_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001212" );
    
    /** <p>GO:0000374.</p> */
    public static final OntClass group_III_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001213" );
    
    /** <p>An exon either containing but not starting with a start codon or containing 
     *  but not ending with a stop codon will be partially coding and partially non 
     *  coding.</p>
     */
    public static final OntClass noncoding_region_of_exon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001214" );
    
    /** <p>An exon containing either a start or stop codon will be partially coding and 
     *  partially non coding.</p>
     */
    public static final OntClass coding_region_of_exon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001215" );
    
    public static final OntClass endonuclease_spliced_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001216" );
    
    public static final OntClass protein_coding_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001217" );
    
    public static final OntClass transgenic_insertion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001218" );
    
    public static final OntClass retrogene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001219" );
    
    /** <p>RNA interference is GO:0016246.</p> */
    public static final OntClass silenced_by_RNA_interference = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001220" );
    
    /** <p>Histone modification is GO:0016570.</p> */
    public static final OntClass silenced_by_histone_modification = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001221" );
    
    /** <p>Histone methylation is GO:0016571.</p> */
    public static final OntClass silenced_by_histone_methylation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001222" );
    
    /** <p>Histone deacetylation is GO:0016573.</p> */
    public static final OntClass silenced_by_histone_deacetylation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001223" );
    
    public static final OntClass gene_silenced_by_RNA_interference = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001224" );
    
    public static final OntClass gene_silenced_by_histone_modification = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001225" );
    
    public static final OntClass gene_silenced_by_histone_methylation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001226" );
    
    public static final OntClass gene_silenced_by_histone_deacetylation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001227" );
    
    public static final OntClass dihydrouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001228" );
    
    /** <p>The free molecule is CHEBI:17802.</p> */
    public static final OntClass pseudouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001229" );
    
    /** <p>The free molecule is CHEBI:17596.</p> */
    public static final OntClass inosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001230" );
    
    /** <p>The free molecule is CHEBI:2274.</p> */
    public static final OntClass seven_methylguanine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001231" );
    
    /** <p>The free molecule is CHEBI:30832.</p> */
    public static final OntClass ribothymidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001232" );
    
    public static final OntClass methylinosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001233" );
    
    public static final OntClass mobile = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001234" );
    
    public static final OntClass replicon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001235" );
    
    public static final OntClass base = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001236" );
    
    /** <p>Probably in the future this will cross reference to Chebi.</p> */
    public static final OntClass amino_acid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001237" );
    
    public static final OntClass major_TSS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001238" );
    
    public static final OntClass minor_TSS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001239" );
    
    public static final OntClass TSS_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001240" );
    
    public static final OntClass encodes_alternate_transcription_start_sites = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001241" );
    
    public static final OntClass miRNA_primary_transcript_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001243" );
    
    public static final OntClass pre_miRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001244" );
    
    public static final OntClass miRNA_stem = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001245" );
    
    public static final OntClass miRNA_loop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001246" );
    
    public static final OntClass synthetic_oligo = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001247" );
    
    public static final OntClass assembly = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001248" );
    
    public static final OntClass fragment_assembly = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001249" );
    
    public static final OntClass fingerprint_map = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001250" );
    
    public static final OntClass STS_map = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001251" );
    
    public static final OntClass RH_map = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001252" );
    
    public static final OntClass sonicate_fragment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001253" );
    
    public static final OntClass polyploid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001254" );
    
    public static final OntClass autopolyploid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001255" );
    
    public static final OntClass allopolyploid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001256" );
    
    public static final OntClass homing_endonuclease_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001257" );
    
    /** <p>Nature. 1986 Oct 16-22;323(6089):640-3.</p> */
    public static final OntClass octamer_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001258" );
    
    public static final OntClass apicoplast_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001259" );
    
    public static final OntClass sequence_collection = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001260" );
    
    /** <p>This feature was requested by Nicole, tracker id 1911479. It is required to 
     *  gather evidence together for annotation. An example would be overlapping ESTs 
     *  that support an mRNA.</p>
     */
    public static final OntClass overlapping_feature_set = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001261" );
    
    public static final OntClass overlapping_EST_set = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001262" );
    
    public static final OntClass ncRNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001263" );
    
    public static final OntClass gRNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001264" );
    
    public static final OntClass miRNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001265" );
    
    public static final OntClass scRNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001266" );
    
    public static final OntClass snoRNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001267" );
    
    public static final OntClass snRNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001268" );
    
    public static final OntClass SRP_RNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001269" );
    
    public static final OntClass tmRNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001271" );
    
    public static final OntClass tRNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001272" );
    
    public static final OntClass modified_adenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001273" );
    
    public static final OntClass modified_inosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001274" );
    
    public static final OntClass modified_cytidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001275" );
    
    public static final OntClass modified_guanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001276" );
    
    public static final OntClass modified_uridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001277" );
    
    public static final OntClass one_methylinosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001278" );
    
    public static final OntClass one_two_prime_O_dimethylinosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001279" );
    
    public static final OntClass two_prime_O_methylinosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001280" );
    
    public static final OntClass three_methylcytidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001281" );
    
    public static final OntClass five_methylcytidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001282" );
    
    public static final OntClass two_prime_O_methylcytidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001283" );
    
    public static final OntClass two_thiocytidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001284" );
    
    public static final OntClass N4_acetylcytidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001285" );
    
    public static final OntClass five_formylcytidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001286" );
    
    public static final OntClass five_two_prime_O_dimethylcytidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001287" );
    
    public static final OntClass N4_acetyl_2_prime_O_methylcytidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001288" );
    
    public static final OntClass lysidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001289" );
    
    public static final OntClass N4_methylcytidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001290" );
    
    public static final OntClass N4_2_prime_O_dimethylcytidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001291" );
    
    public static final OntClass five_hydroxymethylcytidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001292" );
    
    public static final OntClass five_formyl_two_prime_O_methylcytidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001293" );
    
    public static final OntClass N4_N4_2_prime_O_trimethylcytidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001294" );
    
    public static final OntClass one_methyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001295" );
    
    public static final OntClass two_methyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001296" );
    
    public static final OntClass N6_methyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001297" );
    
    public static final OntClass two_prime_O_methyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001298" );
    
    public static final OntClass two_methylthio_N6_methyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001299" );
    
    public static final OntClass N6_isopentenyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001300" );
    
    public static final OntClass two_methylthio_N6_isopentenyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001301" );
    
    public static final OntClass N6_cis_hydroxyisopentenyl_adenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001302" );
    
    public static final OntClass two_methylthio_N6_cis_hydroxyisopentenyl_adenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001303" );
    
    public static final OntClass N6_glycinylcarbamoyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001304" );
    
    public static final OntClass N6_threonylcarbamoyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001305" );
    
    public static final OntClass two_methylthio_N6_threonyl_carbamoyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001306" );
    
    public static final OntClass N6_methyl_N6_threonylcarbamoyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001307" );
    
    public static final OntClass N6_hydroxynorvalylcarbamoyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001308" );
    
    public static final OntClass two_methylthio_N6_hydroxynorvalyl_carbamoyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001309" );
    
    public static final OntClass two_prime_O_ribosyladenosine_phosphate = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001310" );
    
    public static final OntClass N6_N6_dimethyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001311" );
    
    public static final OntClass N6_2_prime_O_dimethyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001312" );
    
    public static final OntClass N6_N6_2_prime_O_trimethyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001313" );
    
    public static final OntClass one_two_prime_O_dimethyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001314" );
    
    public static final OntClass N6_acetyladenosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001315" );
    
    public static final OntClass seven_deazaguanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001316" );
    
    public static final OntClass queuosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001317" );
    
    public static final OntClass epoxyqueuosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001318" );
    
    public static final OntClass galactosyl_queuosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001319" );
    
    public static final OntClass mannosyl_queuosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001320" );
    
    public static final OntClass seven_cyano_seven_deazaguanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001321" );
    
    public static final OntClass seven_aminomethyl_seven_deazaguanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001322" );
    
    public static final OntClass archaeosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001323" );
    
    public static final OntClass one_methylguanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001324" );
    
    public static final OntClass N2_methylguanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001325" );
    
    public static final OntClass seven_methylguanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001326" );
    
    public static final OntClass two_prime_O_methylguanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001327" );
    
    public static final OntClass N2_N2_dimethylguanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001328" );
    
    public static final OntClass N2_2_prime_O_dimethylguanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001329" );
    
    public static final OntClass N2_N2_2_prime_O_trimethylguanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001330" );
    
    public static final OntClass two_prime_O_ribosylguanosine_phosphate = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001331" );
    
    public static final OntClass wybutosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001332" );
    
    public static final OntClass peroxywybutosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001333" );
    
    public static final OntClass hydroxywybutosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001334" );
    
    public static final OntClass undermodified_hydroxywybutosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001335" );
    
    public static final OntClass wyosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001336" );
    
    public static final OntClass methylwyosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001337" );
    
    public static final OntClass N2_7_dimethylguanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001338" );
    
    public static final OntClass N2_N2_7_trimethylguanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001339" );
    
    public static final OntClass one_two_prime_O_dimethylguanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001340" );
    
    public static final OntClass four_demethylwyosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001341" );
    
    public static final OntClass isowyosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001342" );
    
    public static final OntClass N2_7_2prirme_O_trimethylguanosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001343" );
    
    public static final OntClass five_methyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001344" );
    
    public static final OntClass two_prime_O_methyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001345" );
    
    public static final OntClass five_two_prime_O_dimethyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001346" );
    
    public static final OntClass one_methylpseudouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001347" );
    
    public static final OntClass two_prime_O_methylpseudouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001348" );
    
    public static final OntClass two_thiouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001349" );
    
    public static final OntClass four_thiouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001350" );
    
    public static final OntClass five_methyl_2_thiouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001351" );
    
    public static final OntClass two_thio_two_prime_O_methyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001352" );
    
    public static final OntClass three_three_amino_three_carboxypropyl_uridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001353" );
    
    public static final OntClass five_hydroxyuridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001354" );
    
    public static final OntClass five_methoxyuridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001355" );
    
    public static final OntClass uridine_five_oxyacetic_acid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001356" );
    
    public static final OntClass uridine_five_oxyacetic_acid_methyl_ester = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001357" );
    
    public static final OntClass five_carboxyhydroxymethyl_uridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001358" );
    
    public static final OntClass five_carboxyhydroxymethyl_uridine_methyl_ester = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001359" );
    
    public static final OntClass five_methoxycarbonylmethyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001360" );
    
    public static final OntClass five_methoxycarbonylmethyl_two_prime_O_methyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001361" );
    
    public static final OntClass five_methoxycarbonylmethyl_two_thiouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001362" );
    
    public static final OntClass five_aminomethyl_two_thiouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001363" );
    
    public static final OntClass five_methylaminomethyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001364" );
    
    public static final OntClass five_methylaminomethyl_two_thiouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001365" );
    
    public static final OntClass five_methylaminomethyl_two_selenouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001366" );
    
    public static final OntClass five_carbamoylmethyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001367" );
    
    public static final OntClass five_carbamoylmethyl_two_prime_O_methyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001368" );
    
    public static final OntClass five_carboxymethylaminomethyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001369" );
    
    public static final OntClass five_carboxymethylaminomethyl_two_prime_O_methyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001370" );
    
    public static final OntClass five_carboxymethylaminomethyl_two_thiouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001371" );
    
    public static final OntClass three_methyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001372" );
    
    public static final OntClass one_methyl_three_three_amino_three_carboxypropyl_pseudouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001373" );
    
    public static final OntClass five_carboxymethyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001374" );
    
    public static final OntClass three_two_prime_O_dimethyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001375" );
    
    public static final OntClass five_methyldihydrouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001376" );
    
    public static final OntClass three_methylpseudouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001377" );
    
    public static final OntClass five_taurinomethyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001378" );
    
    public static final OntClass five_taurinomethyl_two_thiouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001379" );
    
    public static final OntClass five_isopentenylaminomethyl_uridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001380" );
    
    public static final OntClass five_isopentenylaminomethyl_two_thiouridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001381" );
    
    public static final OntClass five_isopentenylaminomethyl_two_prime_O_methyluridine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001382" );
    
    public static final OntClass histone_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001383" );
    
    public static final OntClass CDS_fragment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001384" );
    
    public static final OntClass modified_amino_acid_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001385" );
    
    public static final OntClass modified_glycine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001386" );
    
    public static final OntClass modified_L_alanine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001387" );
    
    public static final OntClass modified_L_asparagine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001388" );
    
    public static final OntClass modified_L_aspartic_acid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001389" );
    
    public static final OntClass modified_L_cysteine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001390" );
    
    public static final OntClass modified_L_glutamic_acid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001391" );
    
    public static final OntClass modified_L_threonine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001392" );
    
    public static final OntClass modified_L_tryptophan = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001393" );
    
    public static final OntClass modified_L_glutamine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001394" );
    
    public static final OntClass modified_L_methionine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001395" );
    
    public static final OntClass modified_L_isoleucine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001396" );
    
    public static final OntClass modified_L_phenylalanine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001397" );
    
    public static final OntClass modified_L_histidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001398" );
    
    public static final OntClass modified_L_serine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001399" );
    
    public static final OntClass modified_L_lysine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001400" );
    
    public static final OntClass modified_L_leucine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001401" );
    
    public static final OntClass modified_L_selenocysteine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001402" );
    
    public static final OntClass modified_L_valine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001403" );
    
    public static final OntClass modified_L_proline = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001404" );
    
    public static final OntClass modified_L_tyrosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001405" );
    
    public static final OntClass modified_L_arginine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001406" );
    
    public static final OntClass peptidyl = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001407" );
    
    public static final OntClass cleaved_for_gpi_anchor_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001408" );
    
    public static final OntClass biomaterial_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001409" );
    
    public static final OntClass experimental_feature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001410" );
    
    public static final OntClass biological_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001411" );
    
    public static final OntClass topologically_defined_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001412" );
    
    public static final OntClass translocation_breakpoint = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001413" );
    
    public static final OntClass insertion_breakpoint = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001414" );
    
    public static final OntClass deletion_breakpoint = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001415" );
    
    public static final OntClass five_prime_flanking_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001416" );
    
    public static final OntClass three_prime_flanking_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001417" );
    
    /** <p>Term requested by the MODencode group.</p> */
    public static final OntClass transcribed_fragment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001418" );
    
    public static final OntClass cis_splice_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001419" );
    
    public static final OntClass trans_splice_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001420" );
    
    public static final OntClass splice_junction = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001421" );
    
    /** <p>MM Young, K Kirshenbaum, KA Dill &amp; S Highsmith. Predicting conformational 
     *  switches in proteins. Protein Science, 1999, 8, 1752-64. K. Kirshenbaum, M.M. 
     *  Young and S. Highsmith. Predicting Allosteric Switches in Myosins. Protein 
     *  Science 8(9):1806-1815. 1999.</p>
     */
    public static final OntClass conformational_switch = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001422" );
    
    public static final OntClass dye_terminator_read = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001423" );
    
    /** <p>An example is a read produced by Roche 454 technology.</p> */
    public static final OntClass pyrosequenced_read = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001424" );
    
    /** <p>An example of this kind of read is one produced by ABI SOLiD.</p> */
    public static final OntClass ligation_based_read = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001425" );
    
    /** <p>An example is a read produced by Illumina technology.</p> */
    public static final OntClass polymerase_synthesis_read = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001426" );
    
    public static final OntClass cis_regulatory_frameshift_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001427" );
    
    /** <p>From tracker [ 2372385 ] expressed_sequence_assembly.</p> */
    public static final OntClass expressed_sequence_assembly = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001428" );
    
    public static final OntClass DNA_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001429" );
    
    public static final OntClass cryptic_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001431" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_polyadenylation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001432" );
    
    public static final OntClass three_prime_RACE_clone = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001433" );
    
    /** <p>Requested by the Trypanosome community.</p> */
    public static final OntClass cassette_pseudogene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001434" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass alanine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001435" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass valine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001436" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass leucine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001437" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass isoleucine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001438" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass proline = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001439" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass tryptophan = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001440" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass phenylalanine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001441" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass methionine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001442" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass glycine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001443" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass serine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001444" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass threonine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001445" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass tyrosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001446" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass cysteine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001447" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass glutamine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001448" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass asparagine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001449" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass lysine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001450" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass arginine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001451" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass histidine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001452" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass aspartic_acid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001453" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass glutamic_acid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001454" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass selenocysteine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001455" );
    
    /** <p>A place holder for a cross product with chebi.</p> */
    public static final OntClass pyrrolysine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001456" );
    
    /** <p>This term was requested by Jeff Bowes, using the tracker, ID = 2594157.</p> */
    public static final OntClass transcribed_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001457" );
    
    /** <p>This term was requested by Jeff Bowes, using the tracker, ID = 2594157.</p> */
    public static final OntClass unigene_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001458" );
    
    public static final OntClass CRISPR = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001459" );
    
    /** <p>See tracker ID 2060908.</p> */
    public static final OntClass insulator_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001460" );
    
    public static final OntClass enhancer_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001461" );
    
    /** <p>See tracker ID: 2138359.</p> */
    public static final OntClass contig_collection = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001462" );
    
    public static final OntClass lincRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001463" );
    
    public static final OntClass UST = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001464" );
    
    public static final OntClass three_prime_UST = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001465" );
    
    public static final OntClass five_prime_UST = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001466" );
    
    public static final OntClass RST = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001467" );
    
    public static final OntClass three_prime_RST = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001468" );
    
    public static final OntClass five_prime_RST = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001469" );
    
    public static final OntClass UST_match = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001470" );
    
    public static final OntClass RST_match = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001471" );
    
    public static final OntClass primer_match = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001472" );
    
    public static final OntClass miRNA_antiguide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001473" );
    
    public static final OntClass trans_splice_junction = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001474" );
    
    public static final OntClass outron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001475" );
    
    public static final OntClass natural_plasmid = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001476" );
    
    public static final OntClass gene_trap_construct = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001477" );
    
    public static final OntClass promoter_trap_construct = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001478" );
    
    public static final OntClass enhancer_trap_construct = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001479" );
    
    public static final OntClass PAC_end = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001480" );
    
    public static final OntClass RAPD = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001481" );
    
    public static final OntClass shadow_enhancer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001482" );
    
    public static final OntClass SNV = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001483" );
    
    /** <p>X element combinatorial repeats contain Tbf1p binding sites, and possible 
     *  functions include a role in telomerase-independent telomere maintenance via 
     *  recombination or as a barrier against transcriptional silencing. These are 
     *  usually present as a combination of one or more of several types of smaller 
     *  elements (designated A, B, C, or D). This term was requested 2009-10-16 by 
     *  Michel Dumontier, tracker id 2880747.</p>
     */
    public static final OntClass X_element_combinatorial_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001484" );
    
    /** <p>This term was requested 2009-10-16 by Michel Dumontier, tracker id 2880747.</p> */
    public static final OntClass Y_prime_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001485" );
    
    public static final OntClass standard_draft = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001486" );
    
    public static final OntClass high_quality_draft = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001487" );
    
    public static final OntClass improved_high_quality_draft = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001488" );
    
    public static final OntClass annotation_directed_improved_draft = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001489" );
    
    public static final OntClass noncontiguous_finished = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001490" );
    
    public static final OntClass finished_genome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001491" );
    
    public static final OntClass intronic_regulatory_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001492" );
    
    /** <p>This term was requested 2009-10-16 by Michel Dumontier, tracker id 2880699.</p> */
    public static final OntClass centromere_DNA_Element_I = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001493" );
    
    /** <p>This term was requested 2009-10-16 by Michel Dumontier, tracker id 2880699.</p> */
    public static final OntClass centromere_DNA_Element_II = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001494" );
    
    /** <p>This term was requested 2009-10-16 by Michel Dumontier, tracker id 2880699.</p> */
    public static final OntClass centromere_DNA_Element_III = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001495" );
    
    /** <p>The repeats are maintained by telomerase and there is generally 300 (+/-) 
     *  75 bp of TG(1-3) at a given end. Telomeric repeats function in completing 
     *  chromosome replication and protecting the ends from degradation and end-to-end 
     *  fusions. This term was requested 2009-10-16 by Michel Dumontier, tracker id 
     *  2880739.</p>
     */
    public static final OntClass telomeric_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001496" );
    
    /** <p>Possible functions include roles in chromosomal segregation, maintenance of 
     *  chromosome stability, recombinational sequestering, or as a barrier to transcriptional 
     *  silencing. This term was requested 2009-10-16 by Michel Dumontier, tracker 
     *  id 2880747. From Janos Demeter: The only region shared by all chromosome ends, 
     *  the X element core sequence is a small conserved element (~475 bp) that contains 
     *  an ARS sequence and in most cases an Abf1p binding site. Between these is 
     *  a GC-rich region nearly identical to the meiosis-specific regulatory sequence 
     *  URS1.</p>
     */
    public static final OntClass X_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001497" );
    
    public static final OntClass YAC_end = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001498" );
    
    /** <p>This terms and children were added to SO in response to tracker request by 
     *  Patrick Chain. The paper Genome Project Standards in a New Era of Sequencing. 
     *  Science October 9th 2009, addresses these terms.</p>
     */
    public static final OntClass whole_genome_sequence_status = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001499" );
    
    public static final OntClass heritable_phenotypic_marker = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001500" );
    
    /** <p>Term requested via tracker ID: 2910829.</p> */
    public static final OntClass peptide_collection = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001501" );
    
    /** <p>Requested by tracker ID: 2902685.</p> */
    public static final OntClass high_identity_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001502" );
    
    /** <p>Ensembl and Vega also use this term name. Requested by Howard Deen of MGI.</p> */
    public static final OntClass processed_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001503" );
    
    public static final OntClass assortment_derived_variation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001504" );
    
    public static final OntClass reference_genome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001505" );
    
    public static final OntClass variant_genome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001506" );
    
    public static final OntClass variant_collection = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001507" );
    
    public static final OntClass alteration_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001508" );
    
    public static final OntClass chromosomal_variation_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001509" );
    
    public static final OntClass intrachromosomal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001510" );
    
    public static final OntClass interchromosomal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001511" );
    
    public static final OntClass insertion_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001512" );
    
    public static final OntClass tandem = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001513" );
    
    public static final OntClass direct = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001514" );
    
    public static final OntClass inverted = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001515" );
    
    public static final OntClass free = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001516" );
    
    public static final OntClass inversion_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001517" );
    
    public static final OntClass pericentric = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001518" );
    
    public static final OntClass paracentric = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001519" );
    
    public static final OntClass translocaton_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001520" );
    
    public static final OntClass reciprocal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001521" );
    
    public static final OntClass insertional = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001522" );
    
    public static final OntClass duplication_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001523" );
    
    public static final OntClass chromosomally_aberrant_genome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001524" );
    
    public static final OntClass assembly_error_correction = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001525" );
    
    public static final OntClass base_call_error_correction = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001526" );
    
    public static final OntClass peptide_localization_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001527" );
    
    public static final OntClass nuclear_localization_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001528" );
    
    public static final OntClass endosomal_localization_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001529" );
    
    public static final OntClass lysosomal_localization_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001530" );
    
    public static final OntClass nuclear_export_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001531" );
    
    public static final OntClass recombination_signal_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001532" );
    
    public static final OntClass cryptic_splice_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001533" );
    
    public static final OntClass nuclear_rim_localization_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001534" );
    
    public static final OntClass p_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001535" );
    
    public static final OntClass functional_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001536" );
    
    public static final OntClass structural_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001537" );
    
    public static final OntClass transcript_function_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001538" );
    
    public static final OntClass translational_product_function_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001539" );
    
    public static final OntClass level_of_transcript_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001540" );
    
    public static final OntClass decreased_transcript_level_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001541" );
    
    public static final OntClass increased_transcript_level_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001542" );
    
    public static final OntClass transcript_processing_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001543" );
    
    public static final OntClass editing_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001544" );
    
    public static final OntClass polyadenylation_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001545" );
    
    public static final OntClass transcript_stability_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001546" );
    
    public static final OntClass decreased_transcript_stability_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001547" );
    
    public static final OntClass increased_transcript_stability_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001548" );
    
    public static final OntClass transcription_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001549" );
    
    public static final OntClass rate_of_transcription_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001550" );
    
    public static final OntClass increased_transcription_rate_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001551" );
    
    public static final OntClass decreased_transcription_rate_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001552" );
    
    public static final OntClass translational_product_level_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001553" );
    
    public static final OntClass polypeptide_function_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001554" );
    
    public static final OntClass decreased_translational_product_level = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001555" );
    
    public static final OntClass increased_translational_product_level = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001556" );
    
    public static final OntClass polypeptide_gain_of_function_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001557" );
    
    public static final OntClass polypeptide_localization_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001558" );
    
    public static final OntClass polypeptide_loss_of_function_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001559" );
    
    public static final OntClass inactive_ligand_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001560" );
    
    public static final OntClass polypeptide_partial_loss_of_function = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001561" );
    
    public static final OntClass polypeptide_post_translational_processing_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001562" );
    
    public static final OntClass copy_number_change = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001563" );
    
    public static final OntClass gene_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001564" );
    
    public static final OntClass gene_fusion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001565" );
    
    /** <p>EBI term: Regulatory region variations - In regulatory region annotated by 
     *  Ensembl.</p>
     */
    public static final OntClass regulatory_region_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001566" );
    
    public static final OntClass stop_retained_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001567" );
    
    public static final OntClass splicing_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001568" );
    
    public static final OntClass cryptic_splice_site_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001569" );
    
    public static final OntClass cryptic_splice_acceptor = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001570" );
    
    public static final OntClass cryptic_splice_donor = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001571" );
    
    public static final OntClass exon_loss_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001572" );
    
    public static final OntClass intron_gain_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001573" );
    
    public static final OntClass splice_acceptor_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001574" );
    
    public static final OntClass splice_donor_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001575" );
    
    public static final OntClass transcript_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001576" );
    
    /** <p>EBI term: Complex InDel - Insertion or deletion that spans an exon/intron 
     *  border or a coding sequence/UTR border.</p>
     */
    public static final OntClass complex_transcript_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001577" );
    
    /** <p>EBI term: Stop lost - In coding sequence, resulting in the loss of a stop 
     *  codon.</p>
     */
    public static final OntClass stop_lost = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001578" );
    
    public static final OntClass transcript_sequence_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001579" );
    
    public static final OntClass coding_sequence_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001580" );
    
    /** <p>This is being used to annotate changes to the first codon of a transcript, 
     *  when the first annotated codon is not to methionine. A variant is predicted 
     *  to change the first amino acid of a translation irrespective of the fact that 
     *  the underlying codon is an AUG. As such for transcripts with an incomplete 
     *  CDS (sequence does not start with an AUG), it is still called.</p>
     */
    public static final OntClass initiator_codon_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001582" );
    
    /** <p>EBI term: Non-synonymous SNPs. SNPs that are located in the coding sequence 
     *  and result in an amino acid change in the encoded peptide sequence. A change 
     *  that causes a non_synonymous_codon can be more than 3 bases - for example 
     *  4 base substitution.</p>
     */
    public static final OntClass missense_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001583" );
    
    public static final OntClass conservative_missense_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001585" );
    
    public static final OntClass non_conservative_missense_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001586" );
    
    /** <p>EBI term: Stop gained - In coding sequence, resulting in the gain of a stop 
     *  codon (i.e. leading to a shortened peptide sequence).</p>
     */
    public static final OntClass stop_gained = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001587" );
    
    /** <p>EBI term:Frameshift variations - In coding sequence, resulting in a frameshift.</p> */
    public static final OntClass frameshift_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001589" );
    
    /** <p>The terminal codon may be the terminator, or in an incomplete transcript the 
     *  last available codon.</p>
     */
    public static final OntClass terminator_codon_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001590" );
    
    public static final OntClass frame_restoring_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001591" );
    
    public static final OntClass minus_1_frameshift_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001592" );
    
    public static final OntClass minus_2_frameshift_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001593" );
    
    public static final OntClass plus_1_frameshift_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001594" );
    
    public static final OntClass plus_2_frameshift_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001595" );
    
    public static final OntClass transcript_secondary_structure_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001596" );
    
    public static final OntClass compensatory_transcript_secondary_structure_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001597" );
    
    public static final OntClass translational_product_structure_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001598" );
    
    public static final OntClass D_polypeptide_structure_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001599" );
    
    public static final OntClass complex_3D_structural_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001600" );
    
    public static final OntClass conformational_change_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001601" );
    
    public static final OntClass complex_change_of_translational_product_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001602" );
    
    public static final OntClass polypeptide_sequence_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001603" );
    
    public static final OntClass amino_acid_deletion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001604" );
    
    public static final OntClass amino_acid_insertion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001605" );
    
    public static final OntClass amino_acid_substitution = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001606" );
    
    public static final OntClass conservative_amino_acid_substitution = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001607" );
    
    public static final OntClass non_conservative_amino_acid_substitution = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001608" );
    
    public static final OntClass elongated_polypeptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001609" );
    
    public static final OntClass elongated_polypeptide_C_terminal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001610" );
    
    public static final OntClass elongated_polypeptide_N_terminal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001611" );
    
    public static final OntClass elongated_in_frame_polypeptide_C_terminal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001612" );
    
    public static final OntClass elongated_out_of_frame_polypeptide_C_terminal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001613" );
    
    public static final OntClass elongated_in_frame_polypeptide_N_terminal_elongation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001614" );
    
    public static final OntClass elongated_out_of_frame_polypeptide_N_terminal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001615" );
    
    public static final OntClass polypeptide_fusion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001616" );
    
    public static final OntClass polypeptide_truncation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001617" );
    
    public static final OntClass inactive_catalytic_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001618" );
    
    /** <p>Within non-coding gene - Located within a gene that does not code for a protein.</p> */
    public static final OntClass non_coding_transcript_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001619" );
    
    /** <p>EBI term: Within mature miRNA - Located within a microRNA.</p> */
    public static final OntClass mature_miRNA_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001620" );
    
    public static final OntClass NMD_transcript_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001621" );
    
    public static final OntClass UTR_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001622" );
    
    /** <p>EBI term: 5prime UTR variations - In 5prime UTR (untranslated region).</p> */
    public static final OntClass _prime_UTR_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001623" );
    
    /** <p>EBI term 3prime UTR variations - In 3prime UTR.</p> */
    public static final OntClass _prime_UTR_variant_CLASS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001624" );
    
    /** <p>EBI term: Partial codon - Located within the final, incomplete codon of a 
     *  transcript with a shortened coding sequence where the end is unknown.</p>
     */
    public static final OntClass incomplete_terminal_codon_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001626" );
    
    /** <p>EBI term: Intronic variations - In intron.</p> */
    public static final OntClass intron_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001627" );
    
    /** <p>EBI term Intergenic variations - More than 5 kb either upstream or downstream 
     *  of a transcript.</p>
     */
    public static final OntClass intergenic_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001628" );
    
    /** <p>EBI term - essential splice site - In the first 2 or the last 2 base pairs 
     *  of an intron. The 5th base is on the donor (5') side of the intron. Updated 
     *  to b in line with Cancer Genome Project at the Sanger.</p>
     */
    public static final OntClass splice_site_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001629" );
    
    /** <p>EBI term: splice site - 1-3 bps into an exon or 3-8 bps into an intron.</p> */
    public static final OntClass splice_region_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001630" );
    
    /** <p>Different groups annotate up and downstream to different lengths. The subtypes 
     *  are specific and are backed up with cross references.</p>
     */
    public static final OntClass upstream_gene_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001631" );
    
    /** <p>Different groups annotate up and downstream to different lengths. The subtypes 
     *  are specific and are backed up with cross references.</p>
     */
    public static final OntClass downstream_gene_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001632" );
    
    /** <p>EBI term Downstream variations - Within 5 kb downstream of the 3prime end 
     *  of a transcript.</p>
     */
    public static final OntClass KB_downstream_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001633" );
    
    public static final OntClass B_downstream_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001634" );
    
    /** <p>EBI term Upstream variations - Within 5 kb upstream of the 5prime end of a 
     *  transcript.</p>
     */
    public static final OntClass KB_upstream_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001635" );
    
    public static final OntClass KB_upstream_variant_CLASS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001636" );
    
    public static final OntClass rRNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001637" );
    
    public static final OntClass piRNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001638" );
    
    public static final OntClass RNase_P_RNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001639" );
    
    public static final OntClass RNase_MRP_RNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001640" );
    
    public static final OntClass lincRNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001641" );
    
    /** <p>Mathematically defined repeat regions are determined without regard to the 
     *  biological origin of the repetitive region. The repeat units of a MDR are 
     *  the overlapping oligomers of size k that were used to for the query. Tools 
     *  that can annotate mathematically defined repeats include Tallymer (Kurtz et 
     *  al 2008, BMC Genomics: 517) and RePS (Wang et al, Genome Res 12(5): 824-831.).</p>
     */
    public static final OntClass mathematically_defined_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001642" );
    
    public static final OntClass telomerase_RNA_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001643" );
    
    public static final OntClass targeting_vector = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001644" );
    
    public static final OntClass genetic_marker = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001645" );
    
    public static final OntClass DArT_marker = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001646" );
    
    public static final OntClass kozak_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001647" );
    
    public static final OntClass nested_transposon_CLASS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001648" );
    
    public static final OntClass nested_repeat_CLASS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001649" );
    
    public static final OntClass inframe_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001650" );
    
    public static final OntClass retinoic_acid_responsive_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001653" );
    
    public static final OntClass nucleotide_to_protein_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001654" );
    
    /** <p>See GO:0000166 : nucleotide binding.</p> */
    public static final OntClass nucleotide_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001655" );
    
    /** <p>See GO:0046872 : metal ion binding.</p> */
    public static final OntClass metal_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001656" );
    
    public static final OntClass ligand_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001657" );
    
    /** <p>Tracker ID: 3052459.</p> */
    public static final OntClass nested_tandem_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001658" );
    
    public static final OntClass promoter_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001659" );
    
    public static final OntClass core_promoter_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001660" );
    
    public static final OntClass RNA_polymerase_II_TATA_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001661" );
    
    public static final OntClass RNA_polymerase_III_TATA_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001662" );
    
    public static final OntClass BREd_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001663" );
    
    public static final OntClass DCE = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001664" );
    
    public static final OntClass DCE_SI = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001665" );
    
    public static final OntClass DCE_SII = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001666" );
    
    public static final OntClass DCE_SIII = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001667" );
    
    public static final OntClass proximal_promoter_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001668" );
    
    public static final OntClass RNApol_II_core_promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001669" );
    
    public static final OntClass distal_promoter_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001670" );
    
    public static final OntClass bacterial_RNApol_promoter_sigma_70 = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001671" );
    
    public static final OntClass bacterial_RNApol_promoter_sigma54 = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001672" );
    
    public static final OntClass minus_12_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001673" );
    
    public static final OntClass minus_24_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001674" );
    
    /** <p>The A box can be found in the promoters of type 1 and type 2 (pol III) so 
     *  sub-typing here allows the part of relationship of the subtypes to remain 
     *  true.</p>
     */
    public static final OntClass A_box_type_1 = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001675" );
    
    /** <p>The A box can be found in the promoters of type 1 and type 2 (pol III) so 
     *  sub-typing here allows the part of relationship of the subtypes to remain 
     *  true.</p>
     */
    public static final OntClass A_box_type_2 = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001676" );
    
    public static final OntClass intermediate_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001677" );
    
    public static final OntClass regulatory_promoter_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001678" );
    
    public static final OntClass transcription_regulatory_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001679" );
    
    public static final OntClass translation_regulatory_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001680" );
    
    public static final OntClass recombination_regulatory_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001681" );
    
    public static final OntClass replication_regulatory_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001682" );
    
    public static final OntClass sequence_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001683" );
    
    public static final OntClass experimental_feature_attribute = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001684" );
    
    public static final OntClass score = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001685" );
    
    public static final OntClass quality_value = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001686" );
    
    public static final OntClass restriction_enzyme_recognition_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001687" );
    
    public static final OntClass restriction_enzyme_cleavage_junction = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001688" );
    
    public static final OntClass five_prime_restriction_enzyme_junction = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001689" );
    
    public static final OntClass three_prime_restriction_enzyme_junction = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001690" );
    
    public static final OntClass blunt_end_restriction_enzyme_cleavage_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001691" );
    
    public static final OntClass sticky_end_restriction_enzyme_cleavage_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001692" );
    
    public static final OntClass blunt_end_restriction_enzyme_cleavage_junction = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001693" );
    
    public static final OntClass single_strand_restriction_enzyme_cleavage_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001694" );
    
    public static final OntClass restriction_enzyme_single_strand_overhang = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001695" );
    
    public static final OntClass experimentally_defined_binding_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001696" );
    
    public static final OntClass ChIP_seq_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001697" );
    
    public static final OntClass ASPE_primer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001698" );
    
    public static final OntClass dCAPS_primer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001699" );
    
    public static final OntClass histone_modification = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001700" );
    
    public static final OntClass histone_methylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001701" );
    
    public static final OntClass histone_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001702" );
    
    public static final OntClass H3K9_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001703" );
    
    public static final OntClass H3K14_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001704" );
    
    public static final OntClass H3K4_monomethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001705" );
    
    public static final OntClass H3K4_trimethylation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001706" );
    
    public static final OntClass H3K9_trimethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001707" );
    
    public static final OntClass H3K27_monomethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001708" );
    
    public static final OntClass H3K27_trimethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001709" );
    
    public static final OntClass H3K79_monomethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001710" );
    
    public static final OntClass H3K79_dimethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001711" );
    
    public static final OntClass H3K79_trimethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001712" );
    
    public static final OntClass H4K20_monomethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001713" );
    
    public static final OntClass H2BK5_monomethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001714" );
    
    /** <p>Term requested via tracker (2981725) by Alan Ruttenberg, April 2010. It has 
     *  been described as both an enhancer and a promoter, so the parent is the more 
     *  general term.</p>
     */
    public static final OntClass ISRE = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001715" );
    
    public static final OntClass histone_ubiqitination_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001716" );
    
    public static final OntClass H2B_ubiquitination_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001717" );
    
    public static final OntClass H3K18_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001718" );
    
    public static final OntClass H3K23_acylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001719" );
    
    public static final OntClass epigenetically_modified_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001720" );
    
    public static final OntClass H3K27_acylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001721" );
    
    public static final OntClass H3K36_monomethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001722" );
    
    public static final OntClass H3K36_dimethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001723" );
    
    public static final OntClass H3K36_trimethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001724" );
    
    public static final OntClass H3K4_dimethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001725" );
    
    public static final OntClass H3K27_dimethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001726" );
    
    public static final OntClass H3K9_monomethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001727" );
    
    public static final OntClass H3K9_dimethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001728" );
    
    public static final OntClass H4K16_acylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001729" );
    
    public static final OntClass H4K5_acylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001730" );
    
    public static final OntClass H4K8_acylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001731" );
    
    public static final OntClass H3K27_methylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001732" );
    
    public static final OntClass H3K36_methylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001733" );
    
    public static final OntClass H3K4_methylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001734" );
    
    public static final OntClass H3K79_methylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001735" );
    
    public static final OntClass H3K9_methylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001736" );
    
    public static final OntClass histone_acylation_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001737" );
    
    public static final OntClass H4K_acylation_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001738" );
    
    /** <p>Requested by flybase, Dec 2010.</p> */
    public static final OntClass gene_with_non_canonical_start_codon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001739" );
    
    /** <p>Requested by flybase, Dec 2010.</p> */
    public static final OntClass gene_with_start_codon_CUG = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001740" );
    
    public static final OntClass pseudogenic_gene_segment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001741" );
    
    public static final OntClass copy_number_gain = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001742" );
    
    public static final OntClass copy_number_loss = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001743" );
    
    public static final OntClass UPD = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001744" );
    
    public static final OntClass maternal_uniparental_disomy = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001745" );
    
    public static final OntClass paternal_uniparental_disomy = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001746" );
    
    /** <p>Requested by John Calley 3125900.</p> */
    public static final OntClass open_chromatin_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001747" );
    
    public static final OntClass SL3_acceptor_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001748" );
    
    public static final OntClass SL4_acceptor_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001749" );
    
    public static final OntClass SL5_acceptor_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001750" );
    
    public static final OntClass SL6_acceptor_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001751" );
    
    public static final OntClass SL7_acceptor_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001752" );
    
    public static final OntClass SL8_acceptor_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001753" );
    
    public static final OntClass SL9_acceptor_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001754" );
    
    public static final OntClass SL10_acceptor_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001755" );
    
    public static final OntClass SL11_acceptor_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001756" );
    
    public static final OntClass SL12_acceptor_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001757" );
    
    public static final OntClass duplicated_pseudogene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001758" );
    
    /** <p>This is different from a non processed pseudogene because the gene was not 
     *  duplicated. An example is the L-gulono-lactone oxidase pseudogene in primates.</p>
     */
    public static final OntClass unitary_pseudogene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001759" );
    
    public static final OntClass non_processed_pseudogene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001760" );
    
    public static final OntClass variant_quality = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001761" );
    
    public static final OntClass variant_origin = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001762" );
    
    public static final OntClass variant_frequency = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001763" );
    
    public static final OntClass unique_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001764" );
    
    public static final OntClass rare_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001765" );
    
    public static final OntClass polymorphic_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001766" );
    
    public static final OntClass common_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001767" );
    
    public static final OntClass fixed_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001768" );
    
    public static final OntClass variant_phenotype = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001769" );
    
    public static final OntClass benign_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001770" );
    
    public static final OntClass disease_associated_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001771" );
    
    public static final OntClass disease_causing_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001772" );
    
    public static final OntClass lethal_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001773" );
    
    public static final OntClass quantitative_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001774" );
    
    public static final OntClass maternal_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001775" );
    
    public static final OntClass paternal_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001776" );
    
    public static final OntClass somatic_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001777" );
    
    public static final OntClass germline_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001778" );
    
    public static final OntClass pedigree_specific_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001779" );
    
    public static final OntClass population_specific_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001780" );
    
    public static final OntClass de_novo_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001781" );
    
    public static final OntClass TF_binding_site_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001782" );
    
    public static final OntClass complex_structural_alteration = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001784" );
    
    public static final OntClass structural_alteration = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001785" );
    
    public static final OntClass loss_of_heterozygosity = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001786" );
    
    public static final OntClass splice_donor_5th_base_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001787" );
    
    public static final OntClass U_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001788" );
    
    public static final OntClass mating_type_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001789" );
    
    public static final OntClass paired_end_fragment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001790" );
    
    public static final OntClass exon_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001791" );
    
    public static final OntClass non_coding_transcript_exon_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001792" );
    
    public static final OntClass clone_end = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001793" );
    
    public static final OntClass point_centromere = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001794" );
    
    public static final OntClass regional_centromere = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001795" );
    
    public static final OntClass regional_centromere_central_core = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001796" );
    
    public static final OntClass centromeric_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001797" );
    
    public static final OntClass regional_centromere_inner_repeat_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001798" );
    
    public static final OntClass regional_centromere_outer_repeat_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001799" );
    
    public static final OntClass tasiRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001800" );
    
    public static final OntClass tasiRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001801" );
    
    /** <p>Term requested by M. Dumontier, June 1 2011.</p> */
    public static final OntClass increased_polyadenylation_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001802" );
    
    /** <p>Term requested by M. Dumontier, June 1 2011.</p> */
    public static final OntClass decreased_polyadenylation_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001803" );
    
    /** <p>Note: PMID:18794354 describes the DDB box, and has lots of alignments, but 
     *  doesn't actually come out with a consensus sequence.</p>
     */
    public static final OntClass DDB_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001804" );
    
    public static final OntClass destruction_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001805" );
    
    public static final OntClass ER_retention_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001806" );
    
    public static final OntClass KEN_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001807" );
    
    public static final OntClass mitochondrial_targeting_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001808" );
    
    public static final OntClass signal_anchor = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001809" );
    
    public static final OntClass PIP_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001810" );
    
    public static final OntClass phosphorylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001811" );
    
    public static final OntClass transmembrane_helix = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001812" );
    
    public static final OntClass vacuolar_sorting_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001813" );
    
    public static final OntClass coding_variant_quality = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001814" );
    
    public static final OntClass synonymous = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001815" );
    
    public static final OntClass non_synonymous = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001816" );
    
    public static final OntClass inframe = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001817" );
    
    public static final OntClass protein_altering_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001818" );
    
    /** <p>EBI term: Synonymous SNPs - In coding sequence, not resulting in an amino 
     *  acid change (i.e. silent mutation). This term is sometimes used synonomously 
     *  with the more general term 'silent mutation', although a silent mutation may 
     *  occur in non coding sequence. The best practice is to annotate to the most 
     *  specific term.</p>
     */
    public static final OntClass synonymous_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001819" );
    
    public static final OntClass inframe_indel = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001820" );
    
    public static final OntClass inframe_insertion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001821" );
    
    public static final OntClass inframe_deletion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001822" );
    
    public static final OntClass conservative_inframe_insertion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001823" );
    
    public static final OntClass disruptive_inframe_insertion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001824" );
    
    public static final OntClass conservative_inframe_deletion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001825" );
    
    public static final OntClass disruptive_inframe_deletion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001826" );
    
    /** <p>Requested by Bayer Cropscience June, 2011.</p> */
    public static final OntClass mRNA_read = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001827" );
    
    public static final OntClass genomic_DNA_read = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001828" );
    
    /** <p>Requested by Bayer Cropscience June, 2011.</p> */
    public static final OntClass mRNA_contig = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001829" );
    
    /** <p>Requested by Bayer Cropscience June, 2011.</p> */
    public static final OntClass AFLP_fragment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001830" );
    
    public static final OntClass protein_hmm_match = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001831" );
    
    public static final OntClass immunoglobulin_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001832" );
    
    public static final OntClass V_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001833" );
    
    public static final OntClass C_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001834" );
    
    public static final OntClass N_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001835" );
    
    public static final OntClass S_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001836" );
    
    /** <p>Requested by the EBI.</p> */
    public static final OntClass mobile_element_insertion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001837" );
    
    /** <p>Requested by the NCBI.</p> */
    public static final OntClass novel_sequence_insertion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001838" );
    
    public static final OntClass CSL_response_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001839" );
    
    public static final OntClass GATA_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001840" );
    
    /** <p>This terms is used by Ensembl and Vega.</p> */
    public static final OntClass polymorphic_pseudogene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001841" );
    
    public static final OntClass AP_1_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001842" );
    
    public static final OntClass CRE = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001843" );
    
    public static final OntClass CuRE = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001844" );
    
    public static final OntClass DRE = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001845" );
    
    public static final OntClass FLEX_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001846" );
    
    public static final OntClass forkhead_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001847" );
    
    public static final OntClass homol_D_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001848" );
    
    public static final OntClass homol_E_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001849" );
    
    public static final OntClass HSE = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001850" );
    
    /** <p>The synonym IDP (GATA) is found in an annotation but un-traced as far as literature 
     *  goes.</p>
     */
    public static final OntClass iron_repressed_GATA_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001851" );
    
    /** <p>Note that this should not be confused with the M-box that has consensus sequence 
     *  CATGTG and is bound by bHLH transcription factors such as MITF.</p>
     */
    public static final OntClass mating_type_M_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001852" );
    
    public static final OntClass androgen_response_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001853" );
    
    public static final OntClass smFISH_probe = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001854" );
    
    public static final OntClass MCB = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001855" );
    
    public static final OntClass CCAAT_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001856" );
    
    public static final OntClass Ace2_UAS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001857" );
    
    public static final OntClass TR_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001858" );
    
    public static final OntClass STREP_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001859" );
    
    public static final OntClass rDNA_intergenic_spacer_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001860" );
    
    public static final OntClass sterol_regulatory_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001861" );
    
    /** <p>paper:PMID:16043634.</p> */
    public static final OntClass GT_dinucleotide_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001862" );
    
    public static final OntClass GTT_trinucleotide_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001863" );
    
    public static final OntClass Sap1_recognition_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001864" );
    
    public static final OntClass CDRE_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001865" );
    
    /** <p>Requested by Bayer Cropscience December, 2011.</p> */
    public static final OntClass BAC_read_contig = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001866" );
    
    /** <p>Requested by Bayer Cropscience December, 2011.</p> */
    public static final OntClass candidate_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001867" );
    
    /** <p>Requested by Bayer Cropscience December, 2011.</p> */
    public static final OntClass positional_candidate_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001868" );
    
    /** <p>Requested by Bayer Cropscience December, 2011.</p> */
    public static final OntClass functional_candidate_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001869" );
    
    public static final OntClass enhancerRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001870" );
    
    public static final OntClass PCB = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001871" );
    
    /** <p>NCBI definition: An orphan rearrangement between chromosomal location observed 
     *  in isolation.</p>
     */
    public static final OntClass rearrangement_breakpoint = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001872" );
    
    public static final OntClass interchromosomal_breakpoint = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001873" );
    
    public static final OntClass intrachromosomal_breakpoint = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001874" );
    
    /** <p>Requested by Bayer Cropscience January, 2012.</p> */
    public static final OntClass unassigned_supercontig = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001875" );
    
    /** <p>Requested by Bayer Cropscience January, 2012.</p> */
    public static final OntClass partial_genomic_sequence_assembly = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001876" );
    
    public static final OntClass lnc_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001877" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass feature_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001878" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass feature_ablation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001879" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass feature_amplification = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001880" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass feature_translocation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001881" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass feature_fusion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001882" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass transcript_translocation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001883" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass regulatory_region_translocation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001884" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass TFBS_translocation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001885" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass transcript_fusion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001886" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass regulatory_region_fusion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001887" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass TFBS_fusion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001888" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass transcript_amplification = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001889" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass transcript_regulatory_region_fusion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001890" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass regulatory_region_amplification = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001891" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass TFBS_amplification = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001892" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass transcript_ablation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001893" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass regulatory_region_ablation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001894" );
    
    /** <p>Created in conjunction with the EBI.</p> */
    public static final OntClass TFBS_ablation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001895" );
    
    public static final OntClass transposable_element_CDS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001896" );
    
    public static final OntClass transposable_element_pseudogene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001897" );
    
    /** <p>For the S. pombe project - requested by Val Wood.</p> */
    public static final OntClass dg_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001898" );
    
    /** <p>For the S. pombe project - requested by Val Wood.</p> */
    public static final OntClass dh_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001899" );
    
    public static final OntClass M26_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001900" );
    
    public static final OntClass AACCCT_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001901" );
    
    public static final OntClass splice_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001902" );
    
    public static final OntClass intronic_lncRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001903" );
    
    public static final OntClass antisense_lncRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001904" );
    
    public static final OntClass regional_centromere_outer_repeat_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001905" );
    
    public static final OntClass feature_truncation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001906" );
    
    public static final OntClass feature_elongation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001907" );
    
    public static final OntClass internal_feature_elongation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001908" );
    
    public static final OntClass frameshift_elongation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001909" );
    
    public static final OntClass frameshift_truncation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001910" );
    
    public static final OntClass copy_number_increase = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001911" );
    
    public static final OntClass copy_number_decrease = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001912" );
    
    /** <p>Requested by Kevin Clancy - invitrogen -May 2012.</p> */
    public static final OntClass bacterial_RNApol_promoter_sigma_ecf = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001913" );
    
    /** <p>Requested by Midori - June 2012.</p> */
    public static final OntClass rDNA_replication_fork_barrier = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001914" );
    
    public static final OntClass transcription_start_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001915" );
    
    public static final OntClass CAGE_tag = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001916" );
    
    public static final OntClass CAGE_cluster = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001917" );
    
    public static final OntClass _methylcytosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001918" );
    
    public static final OntClass _methylcytosine_CLASS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001919" );
    
    public static final OntClass N6_methyladenine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001920" );
    
    /** <p>Requested by Bayer Cropscience, October, 2012.</p> */
    public static final OntClass mitochondrial_contig = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001921" );
    
    public static final OntClass mitochondrial_supercontig = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001922" );
    
    /** <p>Telomeric transcription has been documented in mammals, birds, fish, plants 
     *  and yeast. Requested by Antonia Lock, October 2012.</p>
     */
    public static final OntClass TERRA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001923" );
    
    /** <p>Telomeric transcription has been documented in mammals, birds, fish, plants 
     *  and yeast. Requested by Antonia Lock, October 2012.</p>
     */
    public static final OntClass ARRET = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001924" );
    
    /** <p>Telomeric transcription has been documented in mammals, birds, fish, plants 
     *  and yeast. Requested by Antonia Lock, October 2012.</p>
     */
    public static final OntClass ARIA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001925" );
    
    /** <p>Telomeric transcription has been documented in mammals, birds, fish, plants 
     *  and yeast. Requested by Antonia Lock, October 2012.</p>
     */
    public static final OntClass anti_ARRET = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001926" );
    
    public static final OntClass telomeric_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001927" );
    
    /** <p>This term is used by Complete Genomics in the structural variant analysis 
     *  files.</p>
     */
    public static final OntClass distal_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001928" );
    
    /** <p>Requested by Bayer Cropscience, October, 2012.</p> */
    public static final OntClass mitochondrial_DNA_read = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001929" );
    
    /** <p>Requested by Bayer Cropscience, October, 2012.</p> */
    public static final OntClass chloroplast_DNA_read = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001930" );
    
    /** <p>Requested by Bayer Cropscience November, 2012.</p> */
    public static final OntClass consensus_gDNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001931" );
    
    public static final OntClass restriction_enzyme_five_prime_single_strand_overhang = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001932" );
    
    public static final OntClass restriction_enzyme_three_prime_single_strand_overhang = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001933" );
    
    public static final OntClass monomeric_repeat = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001934" );
    
    public static final OntClass H3K20_trimethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001935" );
    
    public static final OntClass H3K36_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001936" );
    
    public static final OntClass H2BK12_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001937" );
    
    public static final OntClass H2AK5_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001938" );
    
    public static final OntClass H4K12_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001939" );
    
    public static final OntClass H2BK120_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001940" );
    
    public static final OntClass H4K91_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001941" );
    
    public static final OntClass H2BK20_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001942" );
    
    public static final OntClass H3K4ac_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001943" );
    
    public static final OntClass H2AK9_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001944" );
    
    public static final OntClass H3K56_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001945" );
    
    public static final OntClass H2BK15_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001946" );
    
    public static final OntClass H3R2_monomethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001947" );
    
    public static final OntClass H3R2_dimethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001948" );
    
    public static final OntClass H4R3_dimethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001949" );
    
    public static final OntClass H4K4_trimethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001950" );
    
    public static final OntClass H3K23_dimethylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001951" );
    
    public static final OntClass promoter_flanking_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001952" );
    
    public static final OntClass restriction_enzyme_assembly_scar = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001953" );
    
    /** <p>Not a great term for annotation, but used to classify the various regions 
     *  related to restriction enzymes.</p>
     */
    public static final OntClass restriction_enzyme_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001954" );
    
    public static final OntClass protein_stability_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001955" );
    
    public static final OntClass protease_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001956" );
    
    public static final OntClass RNA_stability_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001957" );
    
    /** <p>Requested by PomBase 3604508.</p> */
    public static final OntClass lariat_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001958" );
    
    public static final OntClass TCT_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001959" );
    
    public static final OntClass _hydroxymethylcytosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001960" );
    
    public static final OntClass _formylcytosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001961" );
    
    public static final OntClass modified_adenine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001962" );
    
    public static final OntClass modified_cytosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001963" );
    
    public static final OntClass modified_guanine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001964" );
    
    public static final OntClass _oxoguanine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001965" );
    
    public static final OntClass _carboxylcytosine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001966" );
    
    public static final OntClass _oxoadenine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001967" );
    
    public static final OntClass coding_transcript_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001968" );
    
    public static final OntClass coding_transcript_intron_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001969" );
    
    public static final OntClass non_coding_transcript_intron_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001970" );
    
    public static final OntClass zinc_finger_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001971" );
    
    public static final OntClass histone_4_acylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001972" );
    
    public static final OntClass histone_3_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001973" );
    
    public static final OntClass CTCF_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001974" );
    
    /** <p>Requested by Jackie Quinn. The sticky restriction sites are different from 
     *  junctions because they include the sequence that is cut, inclusive of the 
     *  five prime junction and the three prime junction.</p>
     */
    public static final OntClass five_prime_sticky_end_restriction_enzyme_cleavage_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001975" );
    
    /** <p>Requested by Jackie Quinn. The sticky restriction sites are different from 
     *  junctions because they include the sequence that is cut, inclusive of the 
     *  five prime junction and the three prime junction.</p>
     */
    public static final OntClass three_prime_sticky_end_restriction_enzyme_cleavage_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001976" );
    
    public static final OntClass ribonuclease_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001977" );
    
    /** <p>Requested by Jackie Quinn for use in synthetic biology.</p> */
    public static final OntClass signature = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001978" );
    
    public static final OntClass RNA_stability_element_CLASS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001979" );
    
    /** <p>A plant specific region.</p> */
    public static final OntClass G_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001980" );
    
    public static final OntClass L_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001981" );
    
    public static final OntClass I_box = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001982" );
    
    /** <p>Requested by Andy Menzies at the Sanger. This isn't necessarily a protein 
     *  coding change. A premature start codon can effect the production of a mature 
     *  protein product by providing a competing translation start point. Some genes 
     *  balance their expression this way, eg THPO requires the presence of a premature 
     *  start to limit expression, its loss leads to Familial thrombocythemia.</p>
     */
    public static final OntClass _prime_UTR_premature_start_codon_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001983" );
    
    public static final OntClass silent_mating_type_cassette_array = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001984" );
    
    /** <p>Requested by Midori Harris, 2013.</p> */
    public static final OntClass Okazaki_fragment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001985" );
    
    /** <p>Requested by Graham Ritchie, EBI/Sanger.</p> */
    public static final OntClass upstream_transcript_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001986" );
    
    /** <p>Requested by Graham Ritchie, EBI/Sanger.</p> */
    public static final OntClass downstream_transcript_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001987" );
    
    public static final OntClass _prime_UTR_premature_start_codon_gain_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001988" );
    
    public static final OntClass _prime_UTR_premature_start_codon_loss_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001989" );
    
    public static final OntClass five_prime_UTR_premature_start_codon_location_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001990" );
    
    /** <p>Requested by Bayer Cropscience September, 2013.</p> */
    public static final OntClass consensus_AFLP_fragment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001991" );
    
    public static final OntClass nonsynonymous_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001992" );
    
    /** <p>Added by Andy Menzies (Sanger).</p> */
    public static final OntClass extended_cis_splice_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001993" );
    
    public static final OntClass intron_base_5 = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001994" );
    
    /** <p>Added by Andy Menzies (Sanger).</p> */
    public static final OntClass extended_intronic_splice_region_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001995" );
    
    public static final OntClass extended_intronic_splice_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001996" );
    
    public static final OntClass subtelomere = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001997" );
    
    public static final OntClass sgRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001998" );
    
    public static final OntClass mating_type_region_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0001999" );
    
    /** <p>Requested by Janos Demeter, SGD.</p> */
    public static final OntClass Y_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002001" );
    
    /** <p>Requested by Janos Demeter, SGD.</p> */
    public static final OntClass Z1_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002002" );
    
    /** <p>Requested by Janos Demeter, SGD.</p> */
    public static final OntClass Z2_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002003" );
    
    public static final OntClass ARS_consensus_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002004" );
    
    /** <p>Requested by Antonia Locke, (Pombe).</p> */
    public static final OntClass DSR_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002005" );
    
    /** <p>This element is bound by Loz1 in S. pombe. The paper does not name the element. 
     *  This term was requested by Midoris Harris, for Pombe.</p>
     */
    public static final OntClass zinc_repressed_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002006" );
    
    public static final OntClass MNV = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002007" );
    
    /** <p>Request from Uma Devi Paila, UVA. Variants in the sites of rare amino acids 
     *  e.g. Selenocysteine. These are important impact terms since a loss of such 
     *  rare amino acids may lead to a loss of function.</p>
     */
    public static final OntClass rare_amino_acid_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002008" );
    
    /** <p>Request from Uma Devi Paila, UVA. Variants in the sites of rare amino acids 
     *  e.g. Selenocysteine. These are important impact terms since a loss of such 
     *  rare amino acids may lead to a loss of function.</p>
     */
    public static final OntClass selenocysteine_loss = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002009" );
    
    /** <p>Request from Uma Devi Paila, UVA. Variants in the sites of rare amino acids 
     *  e.g. Selenocysteine. These are important impact terms since a loss of such 
     *  rare amino acids may lead to a loss of function.</p>
     */
    public static final OntClass pyrrolysine_loss = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002010" );
    
    /** <p>Requested by Pablo Cingolani, for use in SnpEff.</p> */
    public static final OntClass intragenic_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002011" );
    
    /** <p>Request from Uma Devi Paila, UVA. This term should not be applied to incomplete 
     *  transcripts.</p>
     */
    public static final OntClass start_lost = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002012" );
    
    public static final OntClass _prime_UTR_truncation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002013" );
    
    public static final OntClass _prime_UTR_elongation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002014" );
    
    public static final OntClass _prime_UTR_truncation_CLASS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002015" );
    
    public static final OntClass _prime_UTR_elongation_CLASS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002016" );
    
    /** <p>Requested by Uma Paila (UVA) for snpEff.</p> */
    public static final OntClass conserved_intergenic_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002017" );
    
    /** <p>Requested by Uma Paila (UVA) for snpEff.</p> */
    public static final OntClass conserved_intron_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002018" );
    
    /** <p>Requested by Uma Paila as this term is annotated by snpEff. This would be 
     *  used for non_AUG start codon annotation.</p>
     */
    public static final OntClass start_retained_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002019" );
    
    /** <p>Requested by Antonia Lock.</p> */
    public static final OntClass boundary_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002020" );
    
    /** <p>Requested by Midori Harris.</p> */
    public static final OntClass mating_type_region_replication_fork_barrier = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002021" );
    
    public static final OntClass priRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002022" );
    
    public static final OntClass multiplexing_sequence_identifier = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002023" );
    
    /** <p>MERGED COMMENT: TARGET COMMENT: Requested by Janos Demeter, SGD. -------------------- 
     *  SOURCE COMMENT: Requested by Janos Demeter, SGD.</p>
     */
    public static final OntClass W_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002024" );
    
    /** <p>comment: An example of this is the Sme2 locus in fission yeast S. pombe, where 
     *  is coincident with an ribonuclear complex termed the "Mei2 dot". This term 
     *  was Requested by Val Wood, PomBase.</p>
     */
    public static final OntClass cis_acting_homologous_chromosome_pairing_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002025" );
    
    /** <p>Requested by Janos Demeter 2014.</p> */
    public static final OntClass intein_encoding_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002026" );
    
    public static final OntClass uORF = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002027" );
    
    public static final OntClass sORF = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002028" );
    
    public static final OntClass tnaORF = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002029" );
    
    public static final OntClass X_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002030" );
    
    public static final OntClass shRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002031" );
    
    /** <p>MoRs are generated from miR hairpins that are longer and can produce two functional 
     *  miR per strand. They are called moRs because they are not located next to 
     *  the loop and thus their biogenesis process is a little different, but functionally, 
     *  they are supposed to act like miRs. It is the same for loRs that are the loop 
     *  fragments, they are generated differently than miRs or moRs but if loaded 
     *  into the risc they are supposed to act the same way miRs do. Requested by 
     *  Thomas Desvignes, Jan 2015.</p>
     */
    public static final OntClass moR = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002032" );
    
    /** <p>MoRs are generated from miR hairpins that are longer and can produce two functional 
     *  miR per strand. They are called moRs because they are not located next to 
     *  the loop and thus their biogenesis process is a little different, but functionally, 
     *  they are supposed to act like miRs. It is the same for loRs that are the loop 
     *  fragments, they are generated differently than miRs or moRs but if loaded 
     *  into the risc they are supposed to act the same way miRs do. Requested by 
     *  Thomas Desvignes, Jan 2015.</p>
     */
    public static final OntClass loR = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002033" );
    
    public static final OntClass miR_encoding_snoRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002034" );
    
    public static final OntClass lncRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002035" );
    
    public static final OntClass miR_encoding_lncRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002036" );
    
    public static final OntClass miR_encoding_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002037" );
    
    public static final OntClass shRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002038" );
    
    public static final OntClass miR_encoding_shRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002039" );
    
    public static final OntClass vaultRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002040" );
    
    public static final OntClass miR_encoding_vaultRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002041" );
    
    public static final OntClass Y_RNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002042" );
    
    public static final OntClass miR_encoding_Y_RNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002043" );
    
    /** <p>Requested by Rama - SGD.</p> */
    public static final OntClass TCS_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002044" );
    
    /** <p>Requested by Rama, SGD.</p> */
    public static final OntClass pheromone_response_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002045" );
    
    /** <p>Requested by Rama, SGD.</p> */
    public static final OntClass FRE = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002046" );
    
    public static final OntClass transcription_pause_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002047" );
    
    public static final OntClass disabled_reading_frame = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002048" );
    
    /** <p>Requested by: Sagar Jain, Richard Scheuermann.</p> */
    public static final OntClass H3K27_acetylation_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002049" );
    
    public static final OntClass constitutive_promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002050" );
    
    public static final OntClass inducible_promoter = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002051" );
    
    /** <p>Requested by Deanna Church.</p> */
    public static final OntClass dominant_negative_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002052" );
    
    public static final OntClass gain_of_function_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002053" );
    
    public static final OntClass loss_of_function_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002054" );
    
    public static final OntClass null_mutation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002055" );
    
    /** <p>Requested by Javier Diez Perez.</p> */
    public static final OntClass intronic_splicing_silencer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002056" );
    
    public static final OntClass intronic_splicing_enhancer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002057" );
    
    /** <p>Requested by Javier Diez Perez.</p> */
    public static final OntClass exonic_splicing_silencer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002058" );
    
    public static final OntClass recombination_enhancer = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0002059" );
    
    public static final OntClass regulatory_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005836" );
    
    public static final OntClass U14_snoRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005837" );
    
    /** <p>Has RNA 2'-O-ribose methylation guide activity (GO:0030561).</p> */
    public static final OntClass methylation_guide_snoRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005841" );
    
    public static final OntClass rRNA_cleavage_RNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005843" );
    
    public static final OntClass exon_of_single_exon_gene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005845" );
    
    public static final OntClass cassette_array_member = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005847" );
    
    public static final OntClass gene_cassette_member = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005848" );
    
    public static final OntClass gene_subarray_member = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005849" );
    
    public static final OntClass primer_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005850" );
    
    /** <p>This would include, for example, a cluster of genes each encoding the major 
     *  ribosomal RNAs and a cluster of histone gene subarrays.</p>
     */
    public static final OntClass gene_array = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005851" );
    
    /** <p>This would include, for example, a cluster of genes encoding different histones.</p> */
    public static final OntClass gene_subarray = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005852" );
    
    /** <p>This would include, for example, the mating type gene cassettes of S. cerevisiae. 
     *  Gene cassettes usually exist as linear sequences as part of a larger DNA molecule, 
     *  such as a chromosome or plasmid.</p>
     */
    public static final OntClass gene_cassette = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005853" );
    
    /** <p>This would include, for example, the arrays of non-functional VSG genes of 
     *  Trypanosomes.</p>
     */
    public static final OntClass gene_cassette_array = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005854" );
    
    public static final OntClass gene_group = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005855" );
    
    public static final OntClass selenocysteine_tRNA_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005856" );
    
    public static final OntClass selenocysteinyl_tRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005857" );
    
    public static final OntClass syntenic_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0005858" );
    
    /** <p>Range.</p> */
    public static final OntClass biochemical_region_of_peptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100001" );
    
    /** <p>Range.</p> */
    public static final OntClass molecular_contact_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100002" );
    
    public static final OntClass intrinsically_unstructured_polypeptide_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100003" );
    
    public static final OntClass catmat_left_handed_three = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100004" );
    
    public static final OntClass catmat_left_handed_four = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100005" );
    
    public static final OntClass catmat_right_handed_three = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100006" );
    
    public static final OntClass catmat_right_handed_four = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100007" );
    
    public static final OntClass alpha_beta_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100008" );
    
    public static final OntClass lipoprotein_signal_peptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100009" );
    
    public static final OntClass no_output = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100010" );
    
    /** <p>Range.</p> */
    public static final OntClass cleaved_peptide_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100011" );
    
    public static final OntClass peptide_coil = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100012" );
    
    /** <p>Range.</p> */
    public static final OntClass hydrophobic_region_of_peptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100013" );
    
    public static final OntClass n_terminal_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100014" );
    
    public static final OntClass c_terminal_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100015" );
    
    public static final OntClass central_hydrophobic_region_of_signal_peptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100016" );
    
    public static final OntClass polypeptide_conserved_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100017" );
    
    public static final OntClass polypeptide_binding_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100018" );
    
    public static final OntClass polypeptide_catalytic_motif = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100019" );
    
    public static final OntClass polypeptide_DNA_contact = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100020" );
    
    public static final OntClass polypeptide_conserved_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_0100021" );
    
    public static final OntClass substitution = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000002" );
    
    public static final OntClass complex_substitution = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000005" );
    
    public static final OntClass point_mutation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000008" );
    
    public static final OntClass transition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000009" );
    
    public static final OntClass pyrimidine_transition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000010" );
    
    public static final OntClass C_to_T_transition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000011" );
    
    public static final OntClass C_to_T_transition_at_pCpG_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000012" );
    
    public static final OntClass T_to_C_transition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000013" );
    
    public static final OntClass purine_transition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000014" );
    
    public static final OntClass A_to_G_transition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000015" );
    
    public static final OntClass G_to_A_transition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000016" );
    
    public static final OntClass transversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000017" );
    
    public static final OntClass pyrimidine_to_purine_transversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000018" );
    
    public static final OntClass C_to_A_transversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000019" );
    
    public static final OntClass C_to_G_transversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000020" );
    
    public static final OntClass T_to_A_transversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000021" );
    
    public static final OntClass T_to_G_transversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000022" );
    
    public static final OntClass purine_to_pyrimidine_transversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000023" );
    
    public static final OntClass A_to_C_transversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000024" );
    
    public static final OntClass A_to_T_transversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000025" );
    
    public static final OntClass G_to_C_transversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000026" );
    
    public static final OntClass G_to_T_transversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000027" );
    
    public static final OntClass intrachromosomal_mutation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000028" );
    
    public static final OntClass chromosomal_deletion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000029" );
    
    public static final OntClass chromosomal_inversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000030" );
    
    public static final OntClass interchromosomal_mutation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000031" );
    
    /** <p>Indels can have a different number of bases than the corresponding reference 
     *  sequence.</p>
     */
    public static final OntClass indel = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000032" );
    
    public static final OntClass duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000035" );
    
    public static final OntClass inversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000036" );
    
    public static final OntClass chromosomal_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000037" );
    
    public static final OntClass intrachromosomal_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000038" );
    
    public static final OntClass direct_tandem_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000039" );
    
    public static final OntClass inverted_tandem_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000040" );
    
    public static final OntClass intrachromosomal_transposition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000041" );
    
    public static final OntClass compound_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000042" );
    
    public static final OntClass Robertsonian_fusion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000043" );
    
    public static final OntClass chromosomal_translocation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000044" );
    
    public static final OntClass ring_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000045" );
    
    public static final OntClass pericentric_inversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000046" );
    
    public static final OntClass paracentric_inversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000047" );
    
    public static final OntClass reciprocal_chromosomal_translocation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000048" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variation_affecting_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000049" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect. Also as there is not change, it is not a good 
     *  ontological term.</p>
     */
    public static final OntClass sequence_variant_causing_no_change_in_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000050" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variation_affecting_coding_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000054" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_initiator_codon_change_in_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000055" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_amino_acid_coding_codon_change_in_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000056" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_synonymous_codon_change_in_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000057" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_non_synonymous_codon_change_in_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000058" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_missense_codon_change_in_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000059" );
    
    /** <p>The exact rules need to be stated, a common set of rules can be derived from 
     *  e.g. BLOSUM62 amino acid distance matrix.</p>
     */
    public static final OntClass sequence_variant_causing_conservative_missense_codon_change_in_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000060" );
    
    /** <p>The exact rules need to be stated, a common set of rules can be derived from 
     *  e.g. BLOSUM62 amino acid distance matrix.</p>
     */
    public static final OntClass sequence_variant_causing_nonconservative_missense_codon_change_in_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000061" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_nonsense_codon_change_in_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000062" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_terminator_codon_change_in_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000063" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variation_affecting_reading_frame = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000064" );
    
    public static final OntClass frameshift_sequence_variation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000065" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_plus_1_frameshift_mutation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000066" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_minus_1_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000067" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_plus_2_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000068" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_minus_2_frameshift = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000069" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_transcript_processing = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000070" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_splicing = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000071" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_splice_donor = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000072" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_splice_acceptor = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000073" );
    
    /** <p>A cryptic splice site is only used when the natural splice site has been disrupted 
     *  by a sequence alteration.</p>
     */
    public static final OntClass sequence_variant_causing_cryptic_splice_activation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000074" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_editing = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000075" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_transcription = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000076" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_decreasing_rate_of_transcription = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000078" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variation_affecting_transcript_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000079" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_increasing_rate_of_transcription = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000080" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_rate_of_transcription = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000081" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_transcript_stability = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000082" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_increasing_transcript_stability = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000083" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_decreasing_transcript_stability = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000084" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variation_affecting_level_of_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000085" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variation_decreasing_level_of_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000086" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variation_increasing_level_of_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000087" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_translational_product = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000088" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect. Also, as there is no change, this is not a good 
     *  ontological term.</p>
     */
    public static final OntClass sequence_variant_causing_no_change_of_translational_product = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000089" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_complex_change_of_translational_product = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000092" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_amino_acid_substitution = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000093" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_conservative_amino_acid_substitution = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000094" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_nonconservative_amino_acid_substitution = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000095" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_amino_acid_insertion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000096" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_amino_acid_deletion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000097" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_polypeptide_truncation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000098" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_polypeptide_elongation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000099" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass mutation_causing_polypeptide_N_terminal_elongation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000100" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass mutation_causing_polypeptide_C_terminal_elongation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000101" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_level_of_translational_product = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000102" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_decreasing_level_of_translation_product = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000103" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_increasing_level_of_translation_product = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000104" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_polypeptide_amino_acid_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000105" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass mutation_causing_inframe_polypeptide_N_terminal_elongation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000106" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass mutation_causing_out_of_frame_polypeptide_N_terminal_elongation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000107" );
    
    public static final OntClass mutaton_causing_inframe_polypeptide_C_terminal_elongation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000108" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass mutation_causing_out_of_frame_polypeptide_C_terminal_elongation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000109" );
    
    public static final OntClass frame_restoring_sequence_variant = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000110" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_3D_structure_of_polypeptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000111" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect. Also as there is no effect, it is not a good 
     *  term.</p>
     */
    public static final OntClass sequence_variant_causing_no_3D_structural_change = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000112" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_complex_3D_structural_change = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000115" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_conformational_change = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000116" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_polypeptide_function = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000117" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_loss_of_function_of_polypeptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000118" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_inactive_ligand_binding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000119" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_inactive_catalytic_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000120" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_polypeptide_localization_change = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000121" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_polypeptide_post_translational_processing_change = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000122" );
    
    public static final OntClass polypeptide_post_translational_processing_affected = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000123" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_partial_loss_of_function_of_polypeptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000124" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_gain_of_function_of_polypeptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000125" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_transcript_secondary_structure = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000126" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_compensatory_transcript_secondary_structure_mutation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000127" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect. Updated after discussion with Peter Taschner 
     *  - Feb 09.</p>
     */
    public static final OntClass sequence_variant_effect = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000132" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_polypeptide_fusion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000134" );
    
    public static final OntClass autosynaptic_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000136" );
    
    public static final OntClass homo_compound_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000138" );
    
    public static final OntClass hetero_compound_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000140" );
    
    public static final OntClass chromosome_fission = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000141" );
    
    public static final OntClass dexstrosynaptic_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000142" );
    
    public static final OntClass laevosynaptic_chromosome = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000143" );
    
    public static final OntClass free_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000144" );
    
    public static final OntClass free_ring_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000145" );
    
    public static final OntClass complex_chromosomal_mutation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000146" );
    
    public static final OntClass deficient_translocation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000147" );
    
    public static final OntClass inversion_cum_translocation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000148" );
    
    public static final OntClass bipartite_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000149" );
    
    public static final OntClass cyclic_translocation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000150" );
    
    public static final OntClass bipartite_inversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000151" );
    
    public static final OntClass uninverted_insertional_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000152" );
    
    public static final OntClass inverted_insertional_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000153" );
    
    public static final OntClass insertional_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000154" );
    
    public static final OntClass interchromosomal_transposition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000155" );
    
    public static final OntClass inverted_interchromosomal_transposition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000156" );
    
    public static final OntClass uninverted_interchromosomal_transposition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000157" );
    
    public static final OntClass inverted_intrachromosomal_transposition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000158" );
    
    public static final OntClass uninverted_intrachromosomal_transposition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000159" );
    
    /** <p>Flag - unknown in the definition.</p> */
    public static final OntClass unoriented_insertional_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000160" );
    
    /** <p>FLAG - term describes an unknown.</p> */
    public static final OntClass unoriented_interchromosomal_transposition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000161" );
    
    /** <p>FLAG - definition describes an unknown.</p> */
    public static final OntClass unoriented_intrachromosomal_transposition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000162" );
    
    public static final OntClass uncharacterised_chromosomal_mutation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000170" );
    
    public static final OntClass deficient_inversion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000171" );
    
    public static final OntClass tandem_duplication = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000173" );
    
    public static final OntClass partially_characterised_chromosomal_mutation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000175" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_affecting_gene_structure = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000180" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_gene_fusion = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000181" );
    
    public static final OntClass chromosome_number_variation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000182" );
    
    public static final OntClass chromosome_structure_variation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000183" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causes_exon_loss = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000184" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causes_intron_gain = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000185" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_cryptic_splice_donor_activation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1000186" );
    
    /** <p>OBSOLETE: This term was deleted as it conflated more than one term. The alteration 
     *  is separate from the effect.</p>
     */
    public static final OntClass sequence_variant_causing_cryptic_splice_acceptor_activation = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001186" );
    
    public static final OntClass alternatively_spliced_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001187" );
    
    public static final OntClass encodes_1_polypeptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001188" );
    
    public static final OntClass encodes_greater_than_1_polypeptide = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001189" );
    
    public static final OntClass encodes_different_polypeptides_different_stop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001190" );
    
    public static final OntClass encodes_overlapping_peptides_different_start = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001191" );
    
    public static final OntClass encodes_disjoint_polypeptides = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001192" );
    
    public static final OntClass encodes_overlapping_polypeptides_different_start_and_stop = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001193" );
    
    public static final OntClass alternatively_spliced_gene_encoding_greater_than_1_polypeptide_coding_regions_overlapping = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001194" );
    
    public static final OntClass encodes_overlapping_peptides = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001195" );
    
    public static final OntClass cryptogene = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001196" );
    
    public static final OntClass dicistronic_primary_transcript = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001197" );
    
    public static final OntClass member_of_regulon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001217" );
    
    public static final OntClass alternatively_spliced_transcript_encoding_greater_than_1_polypeptide_different_start_codon_different_stop_codon_coding_regions_non_overlapping = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001244" );
    
    public static final OntClass CDS_independently_known = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001246" );
    
    public static final OntClass orphan_CDS = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001247" );
    
    public static final OntClass CDS_supported_by_domain_match_data = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001249" );
    
    public static final OntClass CDS_supported_by_sequence_similarity_data = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001251" );
    
    public static final OntClass CDS_predicted = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001254" );
    
    public static final OntClass status_of_coding_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001255" );
    
    public static final OntClass CDS_supported_by_EST_or_cDNA_data = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001259" );
    
    public static final OntClass internal_Shine_Dalgarno_sequence = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001260" );
    
    public static final OntClass recoded_mRNA = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001261" );
    
    public static final OntClass minus_1_translationally_frameshifted = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001262" );
    
    public static final OntClass plus_1_translationally_frameshifted = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001263" );
    
    public static final OntClass mRNA_recoded_by_translational_bypass = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001264" );
    
    public static final OntClass mRNA_recoded_by_codon_redefinition = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001265" );
    
    public static final OntClass stop_codon_redefinition_as_selenocysteine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001266" );
    
    public static final OntClass stop_codon_readthrough = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001267" );
    
    public static final OntClass recoding_stimulatory_region = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001268" );
    
    public static final OntClass four_bp_start_codon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001269" );
    
    public static final OntClass stop_codon_redefinition_as_pyrrolysine = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001270" );
    
    /** <p>Intron characteristic of tRNA genes; splices by an endonuclease-ligase mediated 
     *  mechanism.</p>
     */
    public static final OntClass archaeal_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001271" );
    
    /** <p>Could be a cross product with Gene ontology, GO:0006388.</p> */
    public static final OntClass tRNA_intron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001272" );
    
    public static final OntClass CTG_start_codon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001273" );
    
    public static final OntClass SECIS_element = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001274" );
    
    public static final OntClass retron = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001275" );
    
    public static final OntClass three_prime_recoding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001277" );
    
    public static final OntClass three_prime_stem_loop_structure = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001279" );
    
    public static final OntClass five_prime_recoding_site = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001280" );
    
    public static final OntClass flanking_three_prime_quadruplet_recoding_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001281" );
    
    public static final OntClass UAG_stop_codon_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001282" );
    
    public static final OntClass UAA_stop_codon_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001283" );
    
    public static final OntClass regulon = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001284" );
    
    public static final OntClass UGA_stop_codon_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001285" );
    
    public static final OntClass three_prime_repeat_recoding_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001286" );
    
    public static final OntClass distant_three_prime_recoding_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001287" );
    
    /** <p>This term does not include the stop codons that are redefined. An example 
     *  would be a stop codon that partially overlapped a frame shifting site would 
     *  be an example stimulatory signal.</p>
     */
    public static final OntClass stop_codon_signal = m_model.createClass( "http://purl.obolibrary.org/obo/SO_1001288" );
    
    public static final OntClass databank_entry = m_model.createClass( "http://purl.obolibrary.org/obo/SO_2000061" );
    
    /** <p>Requested by tracker 2021594, July 2008, by Alex.</p> */
    public static final OntClass gene_segment = m_model.createClass( "http://purl.obolibrary.org/obo/SO_3000000" );
    
}
